// cloudflare/resources/api-gateway/operations/labels.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelBulkUpdateResponsesSinglePage = exports.LabelBulkDeleteResponsesSinglePage = exports.LabelBulkCreateResponsesSinglePage = exports.Labels = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class Labels extends resource_1.APIResource {
    /**
     * Attach label(s) on an operation in endpoint management
     *
     * @example
     * ```ts
     * const label =
     *   await client.apiGateway.operations.labels.create(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    create(operationId, params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/api_gateway/operations/${operationId}/labels`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Replace label(s) on an operation in endpoint management
     *
     * @example
     * ```ts
     * const label =
     *   await client.apiGateway.operations.labels.update(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    update(operationId, params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/api_gateway/operations/${operationId}/labels`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Remove label(s) on an operation in endpoint management
     *
     * @example
     * ```ts
     * const label =
     *   await client.apiGateway.operations.labels.delete(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(operationId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/api_gateway/operations/${operationId}/labels`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Bulk attach label(s) on operation(s) in endpoint management
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const labelBulkCreateResponse of client.apiGateway.operations.labels.bulkCreate(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     selector: {
     *       include: {
     *         operation_ids: [
     *           'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *         ],
     *       },
     *     },
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkCreate(params, options) {
        const { zone_id, ...body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/operations/labels`, LabelBulkCreateResponsesSinglePage, { body, method: 'post', ...options });
    }
    /**
     * Bulk remove label(s) on operation(s) in endpoint management
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const labelBulkDeleteResponse of client.apiGateway.operations.labels.bulkDelete(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkDelete(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/operations/labels`, LabelBulkDeleteResponsesSinglePage, { method: 'delete', ...options });
    }
    /**
     * Bulk replace label(s) on operation(s) in endpoint management
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const labelBulkUpdateResponse of client.apiGateway.operations.labels.bulkUpdate(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     managed: { labels: ['login'] },
     *     selector: {
     *       include: {
     *         operation_ids: [
     *           'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *         ],
     *       },
     *     },
     *     user: { labels: ['login'] },
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkUpdate(params, options) {
        const { zone_id, ...body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/operations/labels`, LabelBulkUpdateResponsesSinglePage, { body, method: 'put', ...options });
    }
}
exports.Labels = Labels;
class LabelBulkCreateResponsesSinglePage extends pagination_1.SinglePage {
}
exports.LabelBulkCreateResponsesSinglePage = LabelBulkCreateResponsesSinglePage;
class LabelBulkDeleteResponsesSinglePage extends pagination_1.SinglePage {
}
exports.LabelBulkDeleteResponsesSinglePage = LabelBulkDeleteResponsesSinglePage;
class LabelBulkUpdateResponsesSinglePage extends pagination_1.SinglePage {
}
exports.LabelBulkUpdateResponsesSinglePage = LabelBulkUpdateResponsesSinglePage;
Labels.LabelBulkCreateResponsesSinglePage = LabelBulkCreateResponsesSinglePage;
Labels.LabelBulkDeleteResponsesSinglePage = LabelBulkDeleteResponsesSinglePage;
Labels.LabelBulkUpdateResponsesSinglePage = LabelBulkUpdateResponsesSinglePage;
