// cloudflare/resources/security-center/insights/audit-logs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { isRequestOptions } from "../../../core.mjs";
import { CloudflareError } from "../../../error.mjs";
import { CursorPagination } from "../../../pagination.mjs";
export class AuditLogs extends APIResource {
    list(params = {}, options) {
        if (isRequestOptions(params)) {
            return this.list({}, params);
        }
        const { account_id, zone_id, ...query } = params;
        if (!account_id && !zone_id) {
            throw new CloudflareError('You must provide either account_id or zone_id.');
        }
        if (account_id && zone_id) {
            throw new CloudflareError('You cannot provide both account_id and zone_id.');
        }
        const { accountOrZone, accountOrZoneId } = account_id ?
            {
                accountOrZone: 'accounts',
                accountOrZoneId: account_id,
            }
            : {
                accountOrZone: 'zones',
                accountOrZoneId: zone_id,
            };
        return this._client.getAPIList(`/${accountOrZone}/${accountOrZoneId}/security-center/insights/audit-log`, AuditLogListResponsesCursorPagination, { query, ...options });
    }
    listByInsight(issueId, params = {}, options) {
        if (isRequestOptions(params)) {
            return this.listByInsight(issueId, {}, params);
        }
        const { account_id, zone_id, ...query } = params;
        if (!account_id && !zone_id) {
            throw new CloudflareError('You must provide either account_id or zone_id.');
        }
        if (account_id && zone_id) {
            throw new CloudflareError('You cannot provide both account_id and zone_id.');
        }
        const { accountOrZone, accountOrZoneId } = account_id ?
            {
                accountOrZone: 'accounts',
                accountOrZoneId: account_id,
            }
            : {
                accountOrZone: 'zones',
                accountOrZoneId: zone_id,
            };
        return this._client.getAPIList(`/${accountOrZone}/${accountOrZoneId}/security-center/insights/${issueId}/audit-log`, AuditLogListByInsightResponsesCursorPagination, { query, ...options });
    }
}
export class AuditLogListResponsesCursorPagination extends CursorPagination {
}
export class AuditLogListByInsightResponsesCursorPagination extends CursorPagination {
}
AuditLogs.AuditLogListResponsesCursorPagination = AuditLogListResponsesCursorPagination;
AuditLogs.AuditLogListByInsightResponsesCursorPagination = AuditLogListByInsightResponsesCursorPagination;
