// cloudflare/resources/security-center/insights/audit-logs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditLogListByInsightResponsesCursorPagination = exports.AuditLogListResponsesCursorPagination = exports.AuditLogs = void 0;
const resource_1 = require("../../../resource.js");
const core_1 = require("../../../core.js");
const error_1 = require("../../../error.js");
const pagination_1 = require("../../../pagination.js");
class AuditLogs extends resource_1.APIResource {
    list(params = {}, options) {
        if ((0, core_1.isRequestOptions)(params)) {
            return this.list({}, params);
        }
        const { account_id, zone_id, ...query } = params;
        if (!account_id && !zone_id) {
            throw new error_1.CloudflareError('You must provide either account_id or zone_id.');
        }
        if (account_id && zone_id) {
            throw new error_1.CloudflareError('You cannot provide both account_id and zone_id.');
        }
        const { accountOrZone, accountOrZoneId } = account_id ?
            {
                accountOrZone: 'accounts',
                accountOrZoneId: account_id,
            }
            : {
                accountOrZone: 'zones',
                accountOrZoneId: zone_id,
            };
        return this._client.getAPIList(`/${accountOrZone}/${accountOrZoneId}/security-center/insights/audit-log`, AuditLogListResponsesCursorPagination, { query, ...options });
    }
    listByInsight(issueId, params = {}, options) {
        if ((0, core_1.isRequestOptions)(params)) {
            return this.listByInsight(issueId, {}, params);
        }
        const { account_id, zone_id, ...query } = params;
        if (!account_id && !zone_id) {
            throw new error_1.CloudflareError('You must provide either account_id or zone_id.');
        }
        if (account_id && zone_id) {
            throw new error_1.CloudflareError('You cannot provide both account_id and zone_id.');
        }
        const { accountOrZone, accountOrZoneId } = account_id ?
            {
                accountOrZone: 'accounts',
                accountOrZoneId: account_id,
            }
            : {
                accountOrZone: 'zones',
                accountOrZoneId: zone_id,
            };
        return this._client.getAPIList(`/${accountOrZone}/${accountOrZoneId}/security-center/insights/${issueId}/audit-log`, AuditLogListByInsightResponsesCursorPagination, { query, ...options });
    }
}
exports.AuditLogs = AuditLogs;
class AuditLogListResponsesCursorPagination extends pagination_1.CursorPagination {
}
exports.AuditLogListResponsesCursorPagination = AuditLogListResponsesCursorPagination;
class AuditLogListByInsightResponsesCursorPagination extends pagination_1.CursorPagination {
}
exports.AuditLogListByInsightResponsesCursorPagination = AuditLogListByInsightResponsesCursorPagination;
AuditLogs.AuditLogListResponsesCursorPagination = AuditLogListResponsesCursorPagination;
AuditLogs.AuditLogListByInsightResponsesCursorPagination = AuditLogListByInsightResponsesCursorPagination;
