// cloudflare/resources/zero-trust/devices/deployment-groups.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class DeploymentGroups extends APIResource {
    /**
     * Creates a new deployment group. Policy IDs must be unique across all deployment
     * groups. This endpoint is in Beta.
     *
     * @example
     * ```ts
     * const deploymentGroup =
     *   await client.zeroTrust.devices.deploymentGroups.create({
     *     account_id: 'account_id',
     *     name: 'Engineering Ring 0',
     *     version_config: [
     *       {
     *         target_environment: 'windows',
     *         version: '2026.5.234.0',
     *       },
     *     ],
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/devices/deployment-groups`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all deployment groups for an account. Use deployment groups to assign
     * target WARP client versions to specific devices. This endpoint is in Beta.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const deploymentGroup of client.zeroTrust.devices.deploymentGroups.list(
     *   { account_id: 'account_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/devices/deployment-groups`, DeploymentGroupsV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes a deployment group. Associated policies no longer apply and devices stop
     * receiving version targets. This endpoint is in Beta.
     *
     * @example
     * ```ts
     * const deploymentGroup =
     *   await client.zeroTrust.devices.deploymentGroups.delete(
     *     'group_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(groupId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/devices/deployment-groups/${groupId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a deployment group. Returns 409 if any newly added policy IDs already
     * belong to another deployment group. This endpoint is in Beta.
     *
     * @example
     * ```ts
     * const deploymentGroup =
     *   await client.zeroTrust.devices.deploymentGroups.edit(
     *     'group_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    edit(groupId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/devices/deployment-groups/${groupId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a single deployment group by its ID. This endpoint is in Beta.
     *
     * @example
     * ```ts
     * const deploymentGroup =
     *   await client.zeroTrust.devices.deploymentGroups.get(
     *     'group_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    get(groupId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/devices/deployment-groups/${groupId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class DeploymentGroupsV4PagePaginationArray extends V4PagePaginationArray {
}
DeploymentGroups.DeploymentGroupsV4PagePaginationArray = DeploymentGroupsV4PagePaginationArray;
