// cloudflare/resources/zero-trust/devices/ip-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPProfilesSinglePage = exports.IPProfiles = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class IPProfiles extends resource_1.APIResource {
    /**
     * Creates a WARP Device IP profile. Currently, only IPv4 Device subnets can be
     * associated.
     *
     * @example
     * ```ts
     * const ipProfile =
     *   await client.zeroTrust.devices.ipProfiles.create({
     *     account_id: 'account_id',
     *     match: 'identity.email == "test@cloudflare.com"',
     *     name: 'IPv4 Cloudflare Source IPs',
     *     precedence: 100,
     *     subnet_id: 'b70ff985-a4ef-4643-bbbc-4a0ed4fc8415',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/devices/ip-profiles`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a WARP Device IP profile. Currently, only IPv4 Device subnets can be
     * associated.
     *
     * @example
     * ```ts
     * const ipProfile =
     *   await client.zeroTrust.devices.ipProfiles.update(
     *     'profile_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    update(profileId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/devices/ip-profiles/${profileId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists WARP Device IP profiles.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ipProfile of client.zeroTrust.devices.ipProfiles.list(
     *   { account_id: 'account_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/devices/ip-profiles`, IPProfilesSinglePage, {
            query,
            ...options,
        });
    }
    /**
     * Delete a WARP Device IP profile.
     *
     * @example
     * ```ts
     * const ipProfile =
     *   await client.zeroTrust.devices.ipProfiles.delete(
     *     'profile_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(profileId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/devices/ip-profiles/${profileId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a single WARP Device IP profile.
     *
     * @example
     * ```ts
     * const ipProfile =
     *   await client.zeroTrust.devices.ipProfiles.get(
     *     'profile_id',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    get(profileId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/devices/ip-profiles/${profileId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.IPProfiles = IPProfiles;
class IPProfilesSinglePage extends pagination_1.SinglePage {
}
exports.IPProfilesSinglePage = IPProfilesSinglePage;
IPProfiles.IPProfilesSinglePage = IPProfilesSinglePage;
