// cloudflare/resources/ddos-protection/advanced-tcp-protection/syn-protection/filters/filters.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterListResponsesV4PagePaginationArray = exports.Filters = void 0;
const resource_1 = require("../../../../../resource.js");
const ItemsAPI = __importStar(require("./items.js"));
const items_1 = require("./items.js");
const pagination_1 = require("../../../../../pagination.js");
class Filters extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a SYN Protection filter for an account.
     *
     * @example
     * ```ts
     * const filter =
     *   await client.DDoSProtection.advancedTCPProtection.synProtection.filters.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       expression:
     *         'ip.dst in { 192.0.2.0/24 198.51.100.0/24 } and tcp.srcport in { 80 443 10000..65535 }',
     *       mode: 'mode',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/filters`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all SYN Protection filters for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const filterListResponse of client.DDoSProtection.advancedTCPProtection.synProtection.filters.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/filters`, FilterListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete all SYN Protection filters for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.DDoSProtection.advancedTCPProtection.synProtection.filters.bulkDelete(
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkDelete(params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/filters`, options);
    }
}
exports.Filters = Filters;
class FilterListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.FilterListResponsesV4PagePaginationArray = FilterListResponsesV4PagePaginationArray;
Filters.FilterListResponsesV4PagePaginationArray = FilterListResponsesV4PagePaginationArray;
Filters.Items = items_1.Items;
