// cloudflare/resources/aisearch/instances/instances.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as JobsAPI from "./jobs.mjs";
import { JobListResponsesV4PagePaginationArray, Jobs, } from "./jobs.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class Instances extends APIResource {
    constructor() {
        super(...arguments);
        this.jobs = new JobsAPI.Jobs(this._client);
    }
    /**
     * Create a new instance.
     *
     * @example
     * ```ts
     * const instance = await client.aiSearch.instances.create({
     *   account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *   id: 'my-ai-search',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/instances`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update instance.
     *
     * @example
     * ```ts
     * const instance = await client.aiSearch.instances.update(
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    update(id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/instances/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List instances.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const instanceListResponse of client.aiSearch.instances.list(
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/instances`, InstanceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete instance.
     *
     * @example
     * ```ts
     * const instance = await client.aiSearch.instances.delete(
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    delete(id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Performs a chat completion request against an AI Search instance, using indexed
     * content as context for generating responses.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.instances.chatCompletions(
     *     'my-ai-search',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       messages: [{ content: 'content', role: 'system' }],
     *     },
     *   );
     * ```
     */
    chatCompletions(id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/instances/${id}/chat/completions`, {
            body,
            ...options,
        });
    }
    /**
     * Read instance.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.instances.read(
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    read(id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Executes a semantic search query against an AI Search instance to find relevant
     * indexed content.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.instances.search(
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    search(id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/instances/${id}/search`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves usage statistics for AI Search instances.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.instances.stats(
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    stats(id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/instances/${id}/stats`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class InstanceListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Instances.InstanceListResponsesV4PagePaginationArray = InstanceListResponsesV4PagePaginationArray;
Instances.Jobs = Jobs;
Instances.JobListResponsesV4PagePaginationArray = JobListResponsesV4PagePaginationArray;
