// cloudflare/resources/aisearch/tokens.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenListResponsesV4PagePaginationArray = exports.Tokens = void 0;
const resource_1 = require("../../resource.js");
const pagination_1 = require("../../pagination.js");
class Tokens extends resource_1.APIResource {
    /**
     * Create a new token.
     *
     * @example
     * ```ts
     * const token = await client.aiSearch.tokens.create({
     *   account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *   cf_api_id: 'a1b2c3d4e5f6',
     *   cf_api_key: 'abc123',
     *   name: 'my-token',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/tokens`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update token.
     *
     * @example
     * ```ts
     * const token = await client.aiSearch.tokens.update(
     *   '62af0db3-c410-40b2-9ee3-0e93f6dd1de0',
     *   {
     *     account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *     cf_api_id: 'a1b2c3d4e5f6',
     *     cf_api_key: 'abc123',
     *     name: 'my-token',
     *   },
     * );
     * ```
     */
    update(id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/tokens/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List tokens.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const tokenListResponse of client.aiSearch.tokens.list(
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/tokens`, TokenListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete token.
     *
     * @example
     * ```ts
     * const token = await client.aiSearch.tokens.delete(
     *   '62af0db3-c410-40b2-9ee3-0e93f6dd1de0',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    delete(id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/tokens/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Read token.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.tokens.read(
     *   '62af0db3-c410-40b2-9ee3-0e93f6dd1de0',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    read(id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/tokens/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Tokens = Tokens;
class TokenListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.TokenListResponsesV4PagePaginationArray = TokenListResponsesV4PagePaginationArray;
Tokens.TokenListResponsesV4PagePaginationArray = TokenListResponsesV4PagePaginationArray;
