// cloudflare/resources/email-sending/subdomains/subdomains.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as DNSAPI from "./dns.mjs";
import { DNS } from "./dns.mjs";
import { SinglePage } from "../../../pagination.mjs";
export class Subdomains extends APIResource {
    constructor() {
        super(...arguments);
        this.dns = new DNSAPI.DNS(this._client);
    }
    /**
     * Creates a new sending subdomain or re-enables sending on an existing subdomain
     * that had it disabled. If zone-level Email Sending has not been enabled yet, the
     * zone flag is automatically set when the entitlement is present.
     *
     * @example
     * ```ts
     * const subdomain =
     *   await client.emailSending.subdomains.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     name: 'sub.example.com',
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/email/sending/subdomains`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all sending-enabled subdomains for the zone.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const subdomainListResponse of client.emailSending.subdomains.list(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/email/sending/subdomains`, SubdomainListResponsesSinglePage, options);
    }
    /**
     * Disables sending on a subdomain and removes its DNS records. If routing is still
     * active on the subdomain, only sending is disabled.
     *
     * @example
     * ```ts
     * const subdomain =
     *   await client.emailSending.subdomains.delete(
     *     'aabbccdd11223344aabbccdd11223344',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(subdomainId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/email/sending/subdomains/${subdomainId}`, options);
    }
    /**
     * Gets information for a specific sending subdomain.
     *
     * @example
     * ```ts
     * const subdomain = await client.emailSending.subdomains.get(
     *   'aabbccdd11223344aabbccdd11223344',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(subdomainId, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/email/sending/subdomains/${subdomainId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class SubdomainListResponsesSinglePage extends SinglePage {
}
Subdomains.SubdomainListResponsesSinglePage = SubdomainListResponsesSinglePage;
Subdomains.DNS = DNS;
