// cloudflare/resources/email-sending/subdomains/subdomains.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubdomainListResponsesSinglePage = exports.Subdomains = void 0;
const resource_1 = require("../../../resource.js");
const DNSAPI = __importStar(require("./dns.js"));
const dns_1 = require("./dns.js");
const pagination_1 = require("../../../pagination.js");
class Subdomains extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.dns = new DNSAPI.DNS(this._client);
    }
    /**
     * Creates a new sending subdomain or re-enables sending on an existing subdomain
     * that had it disabled. If zone-level Email Sending has not been enabled yet, the
     * zone flag is automatically set when the entitlement is present.
     *
     * @example
     * ```ts
     * const subdomain =
     *   await client.emailSending.subdomains.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     name: 'sub.example.com',
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/email/sending/subdomains`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all sending-enabled subdomains for the zone.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const subdomainListResponse of client.emailSending.subdomains.list(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/email/sending/subdomains`, SubdomainListResponsesSinglePage, options);
    }
    /**
     * Disables sending on a subdomain and removes its DNS records. If routing is still
     * active on the subdomain, only sending is disabled.
     *
     * @example
     * ```ts
     * const subdomain =
     *   await client.emailSending.subdomains.delete(
     *     'aabbccdd11223344aabbccdd11223344',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(subdomainId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/email/sending/subdomains/${subdomainId}`, options);
    }
    /**
     * Gets information for a specific sending subdomain.
     *
     * @example
     * ```ts
     * const subdomain = await client.emailSending.subdomains.get(
     *   'aabbccdd11223344aabbccdd11223344',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(subdomainId, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/email/sending/subdomains/${subdomainId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Subdomains = Subdomains;
class SubdomainListResponsesSinglePage extends pagination_1.SinglePage {
}
exports.SubdomainListResponsesSinglePage = SubdomainListResponsesSinglePage;
Subdomains.SubdomainListResponsesSinglePage = SubdomainListResponsesSinglePage;
Subdomains.DNS = dns_1.DNS;
