// cloudflare/resources/aisearch/namespaces/namespaces.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceListResponsesV4PagePaginationArray = exports.Namespaces = void 0;
const resource_1 = require("../../../resource.js");
const InstancesAPI = __importStar(require("./instances/instances.js"));
const instances_1 = require("./instances/instances.js");
const pagination_1 = require("../../../pagination.js");
class Namespaces extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.instances = new InstancesAPI.Instances(this._client);
    }
    /**
     * Create a new namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.create({
     *   account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *   name: 'name',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.update(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    update(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/namespaces/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List namespaces.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const namespaceListResponse of client.aiSearch.namespaces.list(
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/namespaces`, NamespaceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.delete(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    delete(name, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/namespaces/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Performs a chat completion request against multiple AI Search instances in
     * parallel, merging retrieved content as context for generating a response.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.chatCompletions(
     *     'my-namespace',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       ai_search_options: { instance_ids: ['my-ai-search'] },
     *       messages: [{ content: 'content', role: 'system' }],
     *     },
     *   );
     * ```
     */
    chatCompletions(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/chat/completions`, {
            body,
            ...options,
        });
    }
    /**
     * Read namespace.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.namespaces.read(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    read(name, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Multi-Instance Search
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.namespaces.search(
     *   'my-namespace',
     *   {
     *     account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *     ai_search_options: { instance_ids: ['my-ai-search'] },
     *   },
     * );
     * ```
     */
    search(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/search`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Namespaces = Namespaces;
class NamespaceListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.NamespaceListResponsesV4PagePaginationArray = NamespaceListResponsesV4PagePaginationArray;
Namespaces.NamespaceListResponsesV4PagePaginationArray = NamespaceListResponsesV4PagePaginationArray;
Namespaces.Instances = instances_1.Instances;
Namespaces.InstanceListResponsesV4PagePaginationArray = instances_1.InstanceListResponsesV4PagePaginationArray;
