// cloudflare/resources/aisearch/namespaces/namespaces.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as InstancesAPI from "./instances/instances.mjs";
import { InstanceListResponsesV4PagePaginationArray, Instances, } from "./instances/instances.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class Namespaces extends APIResource {
    constructor() {
        super(...arguments);
        this.instances = new InstancesAPI.Instances(this._client);
    }
    /**
     * Create a new namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.create({
     *   account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *   name: 'name',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.update(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    update(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/namespaces/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List namespaces.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const namespaceListResponse of client.aiSearch.namespaces.list(
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/namespaces`, NamespaceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete namespace.
     *
     * @example
     * ```ts
     * const namespace = await client.aiSearch.namespaces.delete(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    delete(name, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/namespaces/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Performs a chat completion request against multiple AI Search instances in
     * parallel, merging retrieved content as context for generating a response.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.chatCompletions(
     *     'my-namespace',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       ai_search_options: { instance_ids: ['my-ai-search'] },
     *       messages: [{ content: 'content', role: 'system' }],
     *     },
     *   );
     * ```
     */
    chatCompletions(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/chat/completions`, {
            body,
            ...options,
        });
    }
    /**
     * Read namespace.
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.namespaces.read(
     *   'production',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * );
     * ```
     */
    read(name, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Multi-Instance Search
     *
     * @example
     * ```ts
     * const response = await client.aiSearch.namespaces.search(
     *   'my-namespace',
     *   {
     *     account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *     ai_search_options: { instance_ids: ['my-ai-search'] },
     *   },
     * );
     * ```
     */
    search(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/search`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
export class NamespaceListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Namespaces.NamespaceListResponsesV4PagePaginationArray = NamespaceListResponsesV4PagePaginationArray;
Namespaces.Instances = Instances;
Namespaces.InstanceListResponsesV4PagePaginationArray = InstanceListResponsesV4PagePaginationArray;
