// cloudflare/resources/aisearch/namespaces/instances/instances.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as ItemsAPI from "./items.mjs";
import { ItemListResponsesV4PagePaginationArray, Items, } from "./items.mjs";
import * as JobsAPI from "./jobs.mjs";
import { JobListResponsesV4PagePaginationArray, Jobs, } from "./jobs.mjs";
import { V4PagePaginationArray } from "../../../../pagination.mjs";
export class Instances extends APIResource {
    constructor() {
        super(...arguments);
        this.jobs = new JobsAPI.Jobs(this._client);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a new instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.create(
     *     'my-namespace',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       id: 'my-ai-search',
     *     },
     *   );
     * ```
     */
    create(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.update(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    update(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List instances.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const instanceListResponse of client.aiSearch.namespaces.instances.list(
     *   'my-namespace',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(name, params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/namespaces/${name}/instances`, InstanceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.delete(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    delete(name, id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Performs a chat completion request against an AI Search instance, using indexed
     * content as context for generating responses.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.chatCompletions(
     *     'my-namespace',
     *     'my-ai-search',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       messages: [{ content: 'content', role: 'system' }],
     *     },
     *   );
     * ```
     */
    chatCompletions(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/chat/completions`, { body, ...options });
    }
    /**
     * Read instance.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.read(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    read(name, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Executes a semantic search query against an AI Search instance to find relevant
     * indexed content.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.search(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    search(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/search`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves usage statistics for AI Search instances.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.stats(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    stats(name, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/stats`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class InstanceListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Instances.InstanceListResponsesV4PagePaginationArray = InstanceListResponsesV4PagePaginationArray;
Instances.Jobs = Jobs;
Instances.JobListResponsesV4PagePaginationArray = JobListResponsesV4PagePaginationArray;
Instances.Items = Items;
Instances.ItemListResponsesV4PagePaginationArray = ItemListResponsesV4PagePaginationArray;
