// cloudflare/resources/aisearch/namespaces/instances/instances.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceListResponsesV4PagePaginationArray = exports.Instances = void 0;
const resource_1 = require("../../../../resource.js");
const ItemsAPI = __importStar(require("./items.js"));
const items_1 = require("./items.js");
const JobsAPI = __importStar(require("./jobs.js"));
const jobs_1 = require("./jobs.js");
const pagination_1 = require("../../../../pagination.js");
class Instances extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.jobs = new JobsAPI.Jobs(this._client);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a new instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.create(
     *     'my-namespace',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       id: 'my-ai-search',
     *     },
     *   );
     * ```
     */
    create(name, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.update(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    update(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List instances.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const instanceListResponse of client.aiSearch.namespaces.instances.list(
     *   'my-namespace',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(name, params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/namespaces/${name}/instances`, InstanceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete instance.
     *
     * @example
     * ```ts
     * const instance =
     *   await client.aiSearch.namespaces.instances.delete(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    delete(name, id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Performs a chat completion request against an AI Search instance, using indexed
     * content as context for generating responses.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.chatCompletions(
     *     'my-namespace',
     *     'my-ai-search',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       messages: [{ content: 'content', role: 'system' }],
     *     },
     *   );
     * ```
     */
    chatCompletions(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/chat/completions`, { body, ...options });
    }
    /**
     * Read instance.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.read(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    read(name, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Executes a semantic search query against an AI Search instance to find relevant
     * indexed content.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.search(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    search(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/search`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves usage statistics for AI Search instances.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.stats(
     *     'my-namespace',
     *     'my-ai-search',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    stats(name, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/stats`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Instances = Instances;
class InstanceListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.InstanceListResponsesV4PagePaginationArray = InstanceListResponsesV4PagePaginationArray;
Instances.InstanceListResponsesV4PagePaginationArray = InstanceListResponsesV4PagePaginationArray;
Instances.Jobs = jobs_1.Jobs;
Instances.JobListResponsesV4PagePaginationArray = jobs_1.JobListResponsesV4PagePaginationArray;
Instances.Items = items_1.Items;
Instances.ItemListResponsesV4PagePaginationArray = items_1.ItemListResponsesV4PagePaginationArray;
