// cloudflare/resources/aisearch/namespaces/instances/items.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemListResponsesV4PagePaginationArray = exports.Items = void 0;
const resource_1 = require("../../../../resource.js");
const Core = __importStar(require("../../../../core.js"));
const pagination_1 = require("../../../../pagination.js");
class Items extends resource_1.APIResource {
    /**
     * Lists indexed items in an AI Search instance.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const itemListResponse of client.aiSearch.namespaces.instances.items.list(
     *   'my-namespace',
     *   'my-ai-search',
     *   { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(name, id, params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items`, ItemListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes a file from a managed AI Search instance and triggers a reindex.
     *
     * @example
     * ```ts
     * const item =
     *   await client.aiSearch.namespaces.instances.items.delete(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    delete(name, id, itemId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists chunks for a specific item in an AI Search instance, including their text
     * content.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.chunks(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    chunks(name, id, itemId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}/chunks`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Creates or updates an indexed item in an AI Search instance.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.createOrUpdate(
     *     'my-namespace',
     *     'my-ai-search',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       key: 'key',
     *       next_action: 'INDEX',
     *     },
     *   );
     * ```
     */
    createOrUpdate(name, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Downloads the raw file content for a specific item from the managed AI Search
     * instance storage.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.download(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    download(name, id, itemId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}/download`, {
            ...options,
            headers: { Accept: 'application/octet-stream', ...options?.headers },
            __binaryResponse: true,
        });
    }
    /**
     * Retrieves a specific indexed item from an AI Search instance.
     *
     * @example
     * ```ts
     * const item =
     *   await client.aiSearch.namespaces.instances.items.get(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    get(name, id, itemId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists processing logs for a specific item in an AI Search instance.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.logs(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     { account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22' },
     *   );
     * ```
     */
    logs(name, id, itemId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}/logs`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Syncs an item to an AI Search instance index.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.sync(
     *     'my-namespace',
     *     'my-ai-search',
     *     'item_id',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       next_action: 'INDEX',
     *     },
     *   );
     * ```
     */
    sync(name, id, itemId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items/${itemId}`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Uploads a file to a managed AI Search instance via multipart/form-data (max
     * 4MB).
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiSearch.namespaces.instances.items.upload(
     *     'my-namespace',
     *     'my-ai-search',
     *     {
     *       account_id: 'c3dc5f0b34a14ff8e1b3ec04895e1b22',
     *       file: { file: fs.createReadStream('path/to/file') },
     *     },
     *   );
     * ```
     */
    upload(name, id, params, options) {
        const { account_id, file } = params;
        return this._client.post(`/accounts/${account_id}/ai-search/namespaces/${name}/instances/${id}/items`, Core.multipartFormRequestOptions({ body: file, ...options }))._thenUnwrap((obj) => obj.result);
    }
}
exports.Items = Items;
class ItemListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.ItemListResponsesV4PagePaginationArray = ItemListResponsesV4PagePaginationArray;
Items.ItemListResponsesV4PagePaginationArray = ItemListResponsesV4PagePaginationArray;
