// cloudflare/resources/schema-validation/schemas.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicSchemasV4PagePaginationArray = exports.Schemas = void 0;
const resource_1 = require("../../resource.js");
const pagination_1 = require("../../pagination.js");
class Schemas extends resource_1.APIResource {
    /**
     * Uploads a new OpenAPI schema for API Shield schema validation. The schema
     * defines expected request/response formats for API endpoints.
     *
     * @example
     * ```ts
     * const publicSchema =
     *   await client.schemaValidation.schemas.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     kind: 'openapi_v3',
     *     name: 'petstore schema',
     *     source: '<schema file contents>',
     *     validation_enabled: true,
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/schema_validation/schemas`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all OpenAPI schemas uploaded to API Shield with pagination support.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const publicSchema of client.schemaValidation.schemas.list(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { zone_id, ...query } = params;
        return this._client.getAPIList(`/zones/${zone_id}/schema_validation/schemas`, PublicSchemasV4PagePaginationArray, { query, ...options });
    }
    /**
     * Permanently removes an uploaded OpenAPI schema from API Shield. Operations using
     * this schema will lose their validation rules.
     *
     * @example
     * ```ts
     * const schema = await client.schemaValidation.schemas.delete(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(schemaId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/schema_validation/schemas/${schemaId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Modifies an existing OpenAPI schema in API Shield, updating the validation rules
     * for associated API operations.
     *
     * @example
     * ```ts
     * const publicSchema =
     *   await client.schemaValidation.schemas.edit(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    edit(schemaId, params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/schema_validation/schemas/${schemaId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Gets the contents and metadata of a specific OpenAPI schema uploaded to API
     * Shield.
     *
     * @example
     * ```ts
     * const publicSchema =
     *   await client.schemaValidation.schemas.get(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(schemaId, params, options) {
        const { zone_id, ...query } = params;
        return this._client.get(`/zones/${zone_id}/schema_validation/schemas/${schemaId}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Schemas = Schemas;
class PublicSchemasV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.PublicSchemasV4PagePaginationArray = PublicSchemasV4PagePaginationArray;
Schemas.PublicSchemasV4PagePaginationArray = PublicSchemasV4PagePaginationArray;
