// cloudflare/resources/workers/observability/destinations.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DestinationListResponsesSinglePage = exports.Destinations = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class Destinations extends resource_1.APIResource {
    /**
     * Create a new Workers Observability Telemetry Destination.
     *
     * @example
     * ```ts
     * const destination =
     *   await client.workers.observability.destinations.create({
     *     account_id: 'account_id',
     *     configuration: {
     *       headers: { foo: 'string' },
     *       logpushDataset: 'opentelemetry-traces',
     *       type: 'logpush',
     *       url: 'url',
     *     },
     *     enabled: true,
     *     name: 'name',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/workers/observability/destinations`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update an existing Workers Observability Telemetry Destination.
     *
     * @example
     * ```ts
     * const destination =
     *   await client.workers.observability.destinations.update(
     *     'slug',
     *     {
     *       account_id: 'account_id',
     *       configuration: {
     *         headers: { foo: 'string' },
     *         type: 'logpush',
     *         url: 'url',
     *       },
     *       enabled: true,
     *     },
     *   );
     * ```
     */
    update(slug, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/workers/observability/destinations/${slug}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List your Workers Observability Telemetry Destinations.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const destinationListResponse of client.workers.observability.destinations.list(
     *   { account_id: 'account_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/workers/observability/destinations`, DestinationListResponsesSinglePage, { query, ...options });
    }
    /**
     * Delete a Workers Observability Telemetry Destination.
     *
     * @example
     * ```ts
     * const destination =
     *   await client.workers.observability.destinations.delete(
     *     'slug',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(slug, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/workers/observability/destinations/${slug}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Destinations = Destinations;
class DestinationListResponsesSinglePage extends pagination_1.SinglePage {
}
exports.DestinationListResponsesSinglePage = DestinationListResponsesSinglePage;
Destinations.DestinationListResponsesSinglePage = DestinationListResponsesSinglePage;
