// cloudflare/resources/r2-data-catalog/r2-data-catalog.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as CredentialsAPI from "./credentials.mjs";
import { Credentials } from "./credentials.mjs";
import * as MaintenanceConfigsAPI from "./maintenance-configs.mjs";
import { MaintenanceConfigs, } from "./maintenance-configs.mjs";
import * as NamespacesAPI from "./namespaces/namespaces.mjs";
import { Namespaces } from "./namespaces/namespaces.mjs";
export class R2DataCatalog extends APIResource {
    constructor() {
        super(...arguments);
        this.maintenanceConfigs = new MaintenanceConfigsAPI.MaintenanceConfigs(this._client);
        this.credentials = new CredentialsAPI.Credentials(this._client);
        this.namespaces = new NamespacesAPI.Namespaces(this._client);
    }
    /**
     * Returns a list of R2 buckets that have been enabled as Apache Iceberg catalogs
     * for the specified account. Each catalog represents an R2 bucket configured to
     * store Iceberg metadata and data files.
     *
     * @example
     * ```ts
     * const r2DataCatalogs = await client.r2DataCatalog.list({
     *   account_id: '0123456789abcdef0123456789abcdef',
     * });
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/r2-catalog`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Disable an R2 bucket as a catalog. This operation deactivates the catalog but
     * preserves existing metadata and data files. The catalog can be re-enabled later.
     *
     * @example
     * ```ts
     * await client.r2DataCatalog.disable('my-data-bucket', {
     *   account_id: '0123456789abcdef0123456789abcdef',
     * });
     * ```
     */
    disable(bucketName, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/r2-catalog/${bucketName}/disable`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Enable an R2 bucket as an Apache Iceberg catalog. This operation creates the
     * necessary catalog infrastructure and activates the bucket for storing Iceberg
     * metadata and data files.
     *
     * @example
     * ```ts
     * const response = await client.r2DataCatalog.enable(
     *   'my-data-bucket',
     *   { account_id: '0123456789abcdef0123456789abcdef' },
     * );
     * ```
     */
    enable(bucketName, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/r2-catalog/${bucketName}/enable`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve detailed information about a specific R2 catalog by bucket name.
     * Returns catalog status, maintenance configuration, and credential status.
     *
     * @example
     * ```ts
     * const r2DataCatalog = await client.r2DataCatalog.get(
     *   'my-data-bucket',
     *   { account_id: '0123456789abcdef0123456789abcdef' },
     * );
     * ```
     */
    get(bucketName, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/r2-catalog/${bucketName}`, options)._thenUnwrap((obj) => obj.result);
    }
}
R2DataCatalog.MaintenanceConfigs = MaintenanceConfigs;
R2DataCatalog.Credentials = Credentials;
R2DataCatalog.Namespaces = Namespaces;
