// cloudflare/resources/r2-data-catalog/r2-data-catalog.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.R2DataCatalog = void 0;
const resource_1 = require("../../resource.js");
const CredentialsAPI = __importStar(require("./credentials.js"));
const credentials_1 = require("./credentials.js");
const MaintenanceConfigsAPI = __importStar(require("./maintenance-configs.js"));
const maintenance_configs_1 = require("./maintenance-configs.js");
const NamespacesAPI = __importStar(require("./namespaces/namespaces.js"));
const namespaces_1 = require("./namespaces/namespaces.js");
class R2DataCatalog extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.maintenanceConfigs = new MaintenanceConfigsAPI.MaintenanceConfigs(this._client);
        this.credentials = new CredentialsAPI.Credentials(this._client);
        this.namespaces = new NamespacesAPI.Namespaces(this._client);
    }
    /**
     * Returns a list of R2 buckets that have been enabled as Apache Iceberg catalogs
     * for the specified account. Each catalog represents an R2 bucket configured to
     * store Iceberg metadata and data files.
     *
     * @example
     * ```ts
     * const r2DataCatalogs = await client.r2DataCatalog.list({
     *   account_id: '0123456789abcdef0123456789abcdef',
     * });
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/r2-catalog`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Disable an R2 bucket as a catalog. This operation deactivates the catalog but
     * preserves existing metadata and data files. The catalog can be re-enabled later.
     *
     * @example
     * ```ts
     * await client.r2DataCatalog.disable('my-data-bucket', {
     *   account_id: '0123456789abcdef0123456789abcdef',
     * });
     * ```
     */
    disable(bucketName, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/r2-catalog/${bucketName}/disable`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Enable an R2 bucket as an Apache Iceberg catalog. This operation creates the
     * necessary catalog infrastructure and activates the bucket for storing Iceberg
     * metadata and data files.
     *
     * @example
     * ```ts
     * const response = await client.r2DataCatalog.enable(
     *   'my-data-bucket',
     *   { account_id: '0123456789abcdef0123456789abcdef' },
     * );
     * ```
     */
    enable(bucketName, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/r2-catalog/${bucketName}/enable`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve detailed information about a specific R2 catalog by bucket name.
     * Returns catalog status, maintenance configuration, and credential status.
     *
     * @example
     * ```ts
     * const r2DataCatalog = await client.r2DataCatalog.get(
     *   'my-data-bucket',
     *   { account_id: '0123456789abcdef0123456789abcdef' },
     * );
     * ```
     */
    get(bucketName, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/r2-catalog/${bucketName}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.R2DataCatalog = R2DataCatalog;
R2DataCatalog.MaintenanceConfigs = maintenance_configs_1.MaintenanceConfigs;
R2DataCatalog.Credentials = credentials_1.Credentials;
R2DataCatalog.Namespaces = namespaces_1.Namespaces;
