// cloudflare/resources/registrar/registrar.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registrar = void 0;
const resource_1 = require("../../resource.js");
const DomainsAPI = __importStar(require("./domains.js"));
const domains_1 = require("./domains.js");
const RegistrationStatusAPI = __importStar(require("./registration-status.js"));
const registration_status_1 = require("./registration-status.js");
const UpdateStatusAPI = __importStar(require("./update-status.js"));
const update_status_1 = require("./update-status.js");
class Registrar extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.domains = new DomainsAPI.Domains(this._client);
        this.registrationStatus = new RegistrationStatusAPI.RegistrationStatus(this._client);
        this.updateStatus = new UpdateStatusAPI.UpdateStatus(this._client);
    }
    /**
     * Performs real-time, authoritative availability checks directly against domain
     * registries. Use this endpoint to verify a domain is available before attempting
     * registration via `POST /registrations`.
     *
     * **Important:** Unlike the Search endpoint, these results are authoritative and
     * reflect current registry status. Always check availability immediately before
     * registration as domain status can change rapidly.
     *
     * **Note:** This endpoint uses POST to accept a list of domains in the request
     * body. It is a read-only operation — it does not create, modify, or reserve any
     * domains.
     *
     * ### Extension support
     *
     * Only domains on extensions supported for programmatic registration by this API
     * can be registered. If you check a domain on an unsupported extension, the
     * response will include `registrable: false` with a `reason` field explaining why:
     *
     * - `extension_not_supported_via_api` — Cloudflare Registrar supports this
     *   extension in the dashboard, but it is not yet available for programmatic
     *   registration via this API. Register via
     *   `https://dash.cloudflare.com/{account_id}/domains/registrations` instead.
     * - `extension_not_supported` — This extension is not supported by Cloudflare
     *   Registrar.
     * - `extension_disallows_registration` — The extension's registry has temporarily
     *   or permanently frozen new registrations. No registrar can register domains on
     *   this extension at this time.
     * - `domain_premium` — The domain is premium priced. Premium registration is not
     *   currently supported by this API.
     * - `domain_unavailable` — The domain is already registered, reserved, or
     *   otherwise not available for registration on a supported extension.
     *
     * The `reason` field is only present when `registrable` is `false`.
     *
     * ### Behavior
     *
     * - Maximum 20 domains per request
     * - Pricing is only returned for domains where `registrable: true`
     * - Results are not cached; each request queries the registry
     *
     * ### Workflow
     *
     * 1. Call this endpoint with domains the user wants to register.
     * 2. For each domain where `registrable: true`, present pricing to the user.
     * 3. If `tier: premium`, note that premium registration is not currently supported
     *    by this API and do not proceed to `POST /registrations`.
     * 4. Proceed to `POST /registrations` only for supported non-premium domains.
     *
     * @example
     * ```ts
     * const response = await client.registrar.check({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   domains: [
     *     'myawesomebrand.com',
     *     'myawesomebrand.net',
     *     'myawesomebrand.org',
     *     'myawesomebrand.app',
     *     'myawesomebrand.dev',
     *   ],
     * });
     * ```
     */
    check(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/registrar/domain-check`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Searches for domain name suggestions based on a keyword, phrase, or partial
     * domain name. Returns a list of potentially available domains with pricing
     * information.
     *
     * **Important:** Results are non-authoritative and based on cached data. Always
     * use the `/domain-check` endpoint to verify real-time availability before
     * attempting registration.
     *
     * Suggestions are scoped to extensions supported for programmatic registration via
     * this API (`POST /registrations`). Domains on unsupported extensions will not
     * appear in results, even if they are available at the registry level.
     *
     * ### Use cases
     *
     * - Brand name discovery (e.g., "acme corp" → acmecorp.com, acmecorp.dev)
     * - Keyword-based suggestions (e.g., "coffee shop" → coffeeshop.com,
     *   mycoffeeshop.net)
     * - Alternative extension discovery (e.g., "example.com" → example.com,
     *   example.app, example.xyz)
     *
     * ### Workflow
     *
     * 1. Call this endpoint with a keyword or domain name.
     * 2. Present suggestions to the user.
     * 3. Call `/domain-check` with the user's chosen domains to confirm real-time
     *    availability and pricing.
     * 4. Proceed to `POST /registrations` only for supported non-premium domains where
     *    the Check response returns `registrable: true`.
     *
     * **Note:** Searching with just a domain extension (e.g., "com" or ".app") is not
     * supported. Provide a keyword or domain name.
     *
     * @example
     * ```ts
     * const response = await client.registrar.search({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   q: 'x',
     * });
     * ```
     */
    search(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/registrar/domain-search`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Registrar = Registrar;
Registrar.Domains = domains_1.Domains;
Registrar.DomainsSinglePage = domains_1.DomainsSinglePage;
Registrar.RegistrationStatus = registration_status_1.RegistrationStatus;
Registrar.UpdateStatus = update_status_1.UpdateStatus;
