// cloudflare/resources/zero-trust/access/users/users.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessUsersV4PagePaginationArray = exports.UserListResponsesV4PagePaginationArray = exports.Users = void 0;
const resource_1 = require("../../../../resource.js");
const ActiveSessionsAPI = __importStar(require("./active-sessions.js"));
const active_sessions_1 = require("./active-sessions.js");
const FailedLoginsAPI = __importStar(require("./failed-logins.js"));
const failed_logins_1 = require("./failed-logins.js");
const LastSeenIdentityAPI = __importStar(require("./last-seen-identity.js"));
const last_seen_identity_1 = require("./last-seen-identity.js");
const pagination_1 = require("../../../../pagination.js");
class Users extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.activeSessions = new ActiveSessionsAPI.ActiveSessions(this._client);
        this.lastSeenIdentity = new LastSeenIdentityAPI.LastSeenIdentity(this._client);
        this.failedLogins = new FailedLoginsAPI.FailedLogins(this._client);
    }
    /**
     * Creates a new user.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.create({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   email: 'jdoe@example.com',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/access/users`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a specific user's name for an account. Requires the user's current email
     * as confirmation (email cannot be changed).
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.update(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     email: 'jdoe@example.com',
     *     name: 'Jane Doe',
     *   },
     * );
     * ```
     */
    update(userId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/access/users/${userId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Gets a list of users for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userListResponse of client.zeroTrust.access.users.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/access/users`, UserListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes a specific user for an account. This will also revoke any active seats
     * and tokens for the user.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.delete(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(userId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/access/users/${userId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Gets a specific user for an account.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.get(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(userId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/access/users/${userId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Users = Users;
class UserListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.UserListResponsesV4PagePaginationArray = UserListResponsesV4PagePaginationArray;
class AccessUsersV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.AccessUsersV4PagePaginationArray = AccessUsersV4PagePaginationArray;
Users.UserListResponsesV4PagePaginationArray = UserListResponsesV4PagePaginationArray;
Users.ActiveSessions = active_sessions_1.ActiveSessions;
Users.ActiveSessionListResponsesSinglePage = active_sessions_1.ActiveSessionListResponsesSinglePage;
Users.LastSeenIdentity = last_seen_identity_1.LastSeenIdentity;
Users.FailedLogins = failed_logins_1.FailedLogins;
Users.FailedLoginListResponsesSinglePage = failed_logins_1.FailedLoginListResponsesSinglePage;
