// cloudflare/resources/zero-trust/access/users/users.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as ActiveSessionsAPI from "./active-sessions.mjs";
import { ActiveSessionListResponsesSinglePage, ActiveSessions, } from "./active-sessions.mjs";
import * as FailedLoginsAPI from "./failed-logins.mjs";
import { FailedLoginListResponsesSinglePage, FailedLogins, } from "./failed-logins.mjs";
import * as LastSeenIdentityAPI from "./last-seen-identity.mjs";
import { LastSeenIdentity } from "./last-seen-identity.mjs";
import { V4PagePaginationArray } from "../../../../pagination.mjs";
export class Users extends APIResource {
    constructor() {
        super(...arguments);
        this.activeSessions = new ActiveSessionsAPI.ActiveSessions(this._client);
        this.lastSeenIdentity = new LastSeenIdentityAPI.LastSeenIdentity(this._client);
        this.failedLogins = new FailedLoginsAPI.FailedLogins(this._client);
    }
    /**
     * Creates a new user.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.create({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   email: 'jdoe@example.com',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/access/users`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a specific user's name for an account. Requires the user's current email
     * as confirmation (email cannot be changed).
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.update(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     email: 'jdoe@example.com',
     *     name: 'Jane Doe',
     *   },
     * );
     * ```
     */
    update(userId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/access/users/${userId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Gets a list of users for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userListResponse of client.zeroTrust.access.users.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/access/users`, UserListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes a specific user for an account. This will also revoke any active seats
     * and tokens for the user.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.delete(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(userId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/access/users/${userId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Gets a specific user for an account.
     *
     * @example
     * ```ts
     * const user = await client.zeroTrust.access.users.get(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(userId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/access/users/${userId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class UserListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
export class AccessUsersV4PagePaginationArray extends V4PagePaginationArray {
}
Users.UserListResponsesV4PagePaginationArray = UserListResponsesV4PagePaginationArray;
Users.ActiveSessions = ActiveSessions;
Users.ActiveSessionListResponsesSinglePage = ActiveSessionListResponsesSinglePage;
Users.LastSeenIdentity = LastSeenIdentity;
Users.FailedLogins = FailedLogins;
Users.FailedLoginListResponsesSinglePage = FailedLoginListResponsesSinglePage;
