// cloudflare/resources/zero-trust/dex/rules.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { V4PagePagination } from "../../../pagination.mjs";
export class Rules extends APIResource {
    /**
     * Create a DEX Rule
     *
     * @example
     * ```ts
     * const rule = await client.zeroTrust.dex.rules.create({
     *   account_id: '01a7362d577a6c3019a474fd6f485823',
     *   match: 'match',
     *   name: 'name',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/dex/rules`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update a DEX Rule
     *
     * @example
     * ```ts
     * const rule = await client.zeroTrust.dex.rules.update(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '01a7362d577a6c3019a474fd6f485823' },
     * );
     * ```
     */
    update(ruleId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/dex/rules/${ruleId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List DEX Rules
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ruleListResponse of client.zeroTrust.dex.rules.list(
     *   {
     *     account_id: '01a7362d577a6c3019a474fd6f485823',
     *     page: 1,
     *     per_page: 1,
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/dex/rules`, RuleListResponsesV4PagePagination, {
            query,
            ...options,
        });
    }
    /**
     * Delete a DEX Rule
     *
     * @example
     * ```ts
     * const rule = await client.zeroTrust.dex.rules.delete(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '01a7362d577a6c3019a474fd6f485823' },
     * );
     * ```
     */
    delete(ruleId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/dex/rules/${ruleId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get details for a DEX Rule
     *
     * @example
     * ```ts
     * const rule = await client.zeroTrust.dex.rules.get(
     *   'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *   { account_id: '01a7362d577a6c3019a474fd6f485823' },
     * );
     * ```
     */
    get(ruleId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/dex/rules/${ruleId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class RuleListResponsesV4PagePagination extends V4PagePagination {
}
Rules.RuleListResponsesV4PagePagination = RuleListResponsesV4PagePagination;
