// cloudflare/resources/realtime-kit/livestreams.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Livestreams = void 0;
const resource_1 = require("../../resource.js");
class Livestreams extends resource_1.APIResource {
    /**
     * Creates a livestream for the given App ID and returns ingest server, stream key,
     * and playback URL. You can pass custom input to the ingest server and stream key,
     * and freely distribute the content using the playback URL on any player that
     * supports HLS/LHLS.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.createIndependentLivestream(
     *     'app_id',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       name: 'prdmmp-xhycsl',
     *     },
     *   );
     * ```
     */
    createIndependentLivestream(appId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/livestreams`, {
            body,
            ...options,
        });
    }
    /**
     * Returns details of all active livestreams for the given livestream ID. Retreive
     * the livestream ID using the `Start livestreaming a meeting` API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getActiveLivestreamsForLivestreamId(
     *     'app_id',
     *     'livestream_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getActiveLivestreamsForLivestreamId(appId, livestreamId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/livestreams/${livestreamId}/active-livestream-session`, options);
    }
    /**
     * Returns details of livestreams associated with the given App ID. It includes
     * livestreams created by your App and RealtimeKit meetings that are livestreamed
     * by your App. If you only want details of livestreams created by your App and not
     * RealtimeKit meetings, you can use the `exclude_meetings` query parameter.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getAllLivestreams(
     *     'app_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getAllLivestreams(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/livestreams`, {
            query,
            ...options,
        });
    }
    /**
     * Returns livestream analytics for the specified time range.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getLivestreamAnalyticsComplete(
     *     'app_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getLivestreamAnalyticsComplete(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/analytics/livestreams/overall`, {
            query,
            ...options,
        });
    }
    /**
     * Returns livestream session details for the given livestream session ID. Retrieve
     * the `livestream_session_id`using the
     * `Fetch livestream session details using a session ID` API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getLivestreamSessionDetailsForSessionId(
     *     'app_id',
     *     'livestream-session-id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getLivestreamSessionDetailsForSessionId(appId, livestreamSessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/livestreams/sessions/${livestreamSessionId}`, options);
    }
    /**
     * Returns details of a livestream with sessions for the given livestream ID.
     * Retreive the livestream ID using the `Start livestreaming a meeting` API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getLivestreamSessionForLivestreamId(
     *     'app_id',
     *     'livestream_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getLivestreamSessionForLivestreamId(appId, livestreamId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/livestreams/${livestreamId}`, {
            query,
            ...options,
        });
    }
    /**
     * Returns details of all active livestreams for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getMeetingActiveLivestreams(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getMeetingActiveLivestreams(appId, meetingId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/active-livestream`, options);
    }
    /**
     * Returns day-wise session and recording analytics data of an App for the
     * specified time range start_date to end_date. If start_date and end_date are not
     * provided, the default time range is set from 30 days ago to the current date.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.getOrgAnalytics(
     *     'app_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getOrgAnalytics(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/analytics/daywise`, {
            query,
            ...options,
        });
    }
    /**
     * Starts livestream of a meeting associated with the given meeting ID. Retreive
     * the meeting ID using the `Create a meeting` API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.startLivestreamingAMeeting(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       name: 'prdmmp-xhycsl',
     *     },
     *   );
     * ```
     */
    startLivestreamingAMeeting(appId, meetingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/livestreams`, { body, ...options });
    }
    /**
     * Stops the active livestream of a meeting associated with the given meeting ID.
     * Retreive the meeting ID using the `Create a meeting` API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.livestreams.stopLivestreamingAMeeting(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    stopLivestreamingAMeeting(appId, meetingId, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/active-livestream/stop`, options);
    }
}
exports.Livestreams = Livestreams;
