// cloudflare/resources/realtime-kit/meetings.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Meetings = void 0;
const resource_1 = require("../../resource.js");
class Meetings extends resource_1.APIResource {
    /**
     * Create a meeting for the given App ID.
     *
     * @example
     * ```ts
     * const meeting = await client.realtimeKit.meetings.create(
     *   'app_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    create(appId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/meetings`, { body, ...options });
    }
    /**
     * Adds a participant to the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.addParticipant(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       custom_participant_id: 'custom_participant_id',
     *       preset_name: 'preset_name',
     *     },
     *   );
     * ```
     */
    addParticipant(appId, meetingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants`, { body, ...options });
    }
    /**
     * Deletes a participant for the given meeting and participant ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.deleteMeetingParticipant(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     'participant_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    deleteMeetingParticipant(appId, meetingId, participantId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants/${participantId}`, options);
    }
    /**
     * Updates a participant's details for the given meeting and participant ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.editParticipant(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     'participant_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    editParticipant(appId, meetingId, participantId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants/${participantId}`, { body, ...options });
    }
    /**
     * Returns all meetings for the given App ID.
     *
     * @example
     * ```ts
     * const meeting = await client.realtimeKit.meetings.get(
     *   'app_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/meetings`, { query, ...options });
    }
    /**
     * Returns a meeting details in an App for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.getMeetingById(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getMeetingById(appId, meetingId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}`, {
            query,
            ...options,
        });
    }
    /**
     * Returns a participant details for the given meeting and participant ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.getMeetingParticipant(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     'participant_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getMeetingParticipant(appId, meetingId, participantId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants/${participantId}`, options);
    }
    /**
     * Returns all participants detail for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.getMeetingParticipants(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getMeetingParticipants(appId, meetingId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants`, { query, ...options });
    }
    /**
     * Regenerates participant's authentication token for the given meeting and
     * participant ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.refreshParticipantToken(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     'participant_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    refreshParticipantToken(appId, meetingId, participantId, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}/participants/${participantId}/token`, options);
    }
    /**
     * Replaces all the details for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.replaceMeetingById(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    replaceMeetingById(appId, meetingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}`, {
            body,
            ...options,
        });
    }
    /**
     * Updates a meeting in an App for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.meetings.updateMeetingById(
     *     'app_id',
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    updateMeetingById(appId, meetingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/realtime/kit/${appId}/meetings/${meetingId}`, {
            body,
            ...options,
        });
    }
}
exports.Meetings = Meetings;
