// cloudflare/resources/realtime-kit/realtime-kit.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as ActiveSessionAPI from "./active-session.mjs";
import { ActiveSession, } from "./active-session.mjs";
import * as AnalyticsAPI from "./analytics.mjs";
import { Analytics } from "./analytics.mjs";
import * as AppsAPI from "./apps.mjs";
import { Apps } from "./apps.mjs";
import * as LivestreamsAPI from "./livestreams.mjs";
import { Livestreams, } from "./livestreams.mjs";
import * as MeetingsAPI from "./meetings.mjs";
import { Meetings, } from "./meetings.mjs";
import * as PresetsAPI from "./presets.mjs";
import { Presets, } from "./presets.mjs";
import * as RecordingsAPI from "./recordings.mjs";
import { Recordings, } from "./recordings.mjs";
import * as SessionsAPI from "./sessions.mjs";
import { Sessions, } from "./sessions.mjs";
import * as WebhooksAPI from "./webhooks.mjs";
import { Webhooks, } from "./webhooks.mjs";
export class RealtimeKit extends APIResource {
    constructor() {
        super(...arguments);
        this.apps = new AppsAPI.Apps(this._client);
        this.meetings = new MeetingsAPI.Meetings(this._client);
        this.presets = new PresetsAPI.Presets(this._client);
        this.sessions = new SessionsAPI.Sessions(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.webhooks = new WebhooksAPI.Webhooks(this._client);
        this.activeSession = new ActiveSessionAPI.ActiveSession(this._client);
        this.livestreams = new LivestreamsAPI.Livestreams(this._client);
        this.analytics = new AnalyticsAPI.Analytics(this._client);
    }
}
RealtimeKit.Apps = Apps;
RealtimeKit.Meetings = Meetings;
RealtimeKit.Presets = Presets;
RealtimeKit.Sessions = Sessions;
RealtimeKit.Recordings = Recordings;
RealtimeKit.Webhooks = Webhooks;
RealtimeKit.ActiveSession = ActiveSession;
RealtimeKit.Livestreams = Livestreams;
RealtimeKit.Analytics = Analytics;
