// cloudflare/resources/realtime-kit/realtime-kit.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RealtimeKit = void 0;
const resource_1 = require("../../resource.js");
const ActiveSessionAPI = __importStar(require("./active-session.js"));
const active_session_1 = require("./active-session.js");
const AnalyticsAPI = __importStar(require("./analytics.js"));
const analytics_1 = require("./analytics.js");
const AppsAPI = __importStar(require("./apps.js"));
const apps_1 = require("./apps.js");
const LivestreamsAPI = __importStar(require("./livestreams.js"));
const livestreams_1 = require("./livestreams.js");
const MeetingsAPI = __importStar(require("./meetings.js"));
const meetings_1 = require("./meetings.js");
const PresetsAPI = __importStar(require("./presets.js"));
const presets_1 = require("./presets.js");
const RecordingsAPI = __importStar(require("./recordings.js"));
const recordings_1 = require("./recordings.js");
const SessionsAPI = __importStar(require("./sessions.js"));
const sessions_1 = require("./sessions.js");
const WebhooksAPI = __importStar(require("./webhooks.js"));
const webhooks_1 = require("./webhooks.js");
class RealtimeKit extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.apps = new AppsAPI.Apps(this._client);
        this.meetings = new MeetingsAPI.Meetings(this._client);
        this.presets = new PresetsAPI.Presets(this._client);
        this.sessions = new SessionsAPI.Sessions(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.webhooks = new WebhooksAPI.Webhooks(this._client);
        this.activeSession = new ActiveSessionAPI.ActiveSession(this._client);
        this.livestreams = new LivestreamsAPI.Livestreams(this._client);
        this.analytics = new AnalyticsAPI.Analytics(this._client);
    }
}
exports.RealtimeKit = RealtimeKit;
RealtimeKit.Apps = apps_1.Apps;
RealtimeKit.Meetings = meetings_1.Meetings;
RealtimeKit.Presets = presets_1.Presets;
RealtimeKit.Sessions = sessions_1.Sessions;
RealtimeKit.Recordings = recordings_1.Recordings;
RealtimeKit.Webhooks = webhooks_1.Webhooks;
RealtimeKit.ActiveSession = active_session_1.ActiveSession;
RealtimeKit.Livestreams = livestreams_1.Livestreams;
RealtimeKit.Analytics = analytics_1.Analytics;
