// cloudflare/resources/realtime-kit/recordings.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
export class Recordings extends APIResource {
    /**
     * Returns the active recording details for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.recordings.getActiveRecordings(
     *     'app_id',
     *     'meeting_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getActiveRecordings(appId, meetingId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/recordings/active-recording/${meetingId}`, options);
    }
    /**
     * Returns details of a recording for the given recording ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.recordings.getOneRecording(
     *     'app_id',
     *     'recording_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getOneRecording(appId, recordingId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/recordings/${recordingId}`, options);
    }
    /**
     * Returns all recordings for an App. If the `meeting_id` parameter is passed,
     * returns all recordings for the given meeting ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.recordings.getRecordings(
     *     'app_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getRecordings(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/recordings`, {
            query,
            ...options,
        });
    }
    /**
     * Pause/Resume/Stop a given recording ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.recordings.pauseResumeStopRecording(
     *     '2a95132c15732412d22c1476fa83f27a',
     *     'recording_id',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       action: 'stop',
     *     },
     *   );
     * ```
     */
    pauseResumeStopRecording(appId, recordingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/realtime/kit/${appId}/recordings/${recordingId}`, {
            body,
            ...options,
        });
    }
    /**
     * Starts recording a meeting. The meeting can be started by an App admin directly,
     * or a participant with permissions to start a recording, based on the type of
     * authorization used.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.recordings.startRecordings(
     *     'app_id',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       audio_config: {
     *         channel: 'stereo',
     *         codec: 'AAC',
     *         export_file: true,
     *       },
     *       file_name_prefix: 'string',
     *       interactive_config: { type: 'ID3' },
     *       max_seconds: 60,
     *       meeting_id: '97440c6a-140b-40a9-9499-b23fd7a3868a',
     *       realtimekit_bucket_config: { enabled: true },
     *       video_config: {
     *         codec: 'H264',
     *         export_file: true,
     *         height: 720,
     *         watermark: {
     *           position: 'left top',
     *           size: { height: 1, width: 1 },
     *           url: 'http://example.com',
     *         },
     *         width: 1280,
     *       },
     *     },
     *   );
     * ```
     */
    startRecordings(appId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/recordings`, {
            body,
            ...options,
        });
    }
    /**
     * Starts a track recording in a meeting. Track recordings consist of "layers".
     * Layers are used to map audio/video tracks in a meeting to output destinations.
     * More information about track recordings is available in the
     * [Track Recordings Guide Page](https://docs.realtime.cloudflare.com/guides/capabilities/recording/recording-overview).
     *
     * @example
     * ```ts
     * await client.realtimeKit.recordings.startTrackRecording(
     *   'app_id',
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     layers: {
     *       default: {
     *         file_name_prefix: 'string',
     *         outputs: [{ type: 'REALTIMEKIT_BUCKET' }],
     *       },
     *       'default-video': {
     *         file_name_prefix: 'string',
     *         outputs: [{ type: 'REALTIMEKIT_BUCKET' }],
     *       },
     *     },
     *     meeting_id: 'string',
     *     max_seconds: 60,
     *   },
     * );
     * ```
     */
    startTrackRecording(appId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/recordings/track`, {
            body,
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
}
