// cloudflare/resources/realtime-kit/sessions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
export class Sessions extends APIResource {
    /**
     * Trigger Summary generation of Transcripts for the session ID.
     *
     * @example
     * ```ts
     * await client.realtimeKit.sessions.generateSummaryOfTranscripts(
     *   'app_id',
     *   'session_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    generateSummaryOfTranscripts(appId, sessionId, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/summary`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Returns details of the given peer ID along with call statistics for the given
     * session ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getParticipantDataFromPeerId(
     *     'app_id',
     *     'peer_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getParticipantDataFromPeerId(appId, peerId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/peer-report/${peerId}`, {
            query,
            ...options,
        });
    }
    /**
     * Returns a URL to download all chat messages of the session ID in CSV format.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionChat(
     *     'app_id',
     *     'session_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionChat(appId, sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/chat`, options);
    }
    /**
     * Returns data of the given session ID including recording details.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionDetails(
     *     'app_id',
     *     'session_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionDetails(appId, sessionId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}`, {
            query,
            ...options,
        });
    }
    /**
     * Returns details of the given participant ID along with call statistics for the
     * given session ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionParticipantDetails(
     *     'app_id',
     *     'session_id',
     *     'participant_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionParticipantDetails(appId, sessionId, participantId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/participants/${participantId}`, { query, ...options });
    }
    /**
     * Returns a list of participants for the given session ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionParticipants(
     *     'app_id',
     *     'session_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionParticipants(appId, sessionId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/participants`, { query, ...options });
    }
    /**
     * Returns a Summary URL to download the Summary of Transcripts for the session ID
     * as plain text.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionSummary(
     *     'app_id',
     *     'session_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionSummary(appId, sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/summary`, options);
    }
    /**
     * Returns a URL to download the transcript for the session ID in CSV format.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessionTranscripts(
     *     'app_id',
     *     'session_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getSessionTranscripts(appId, sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions/${sessionId}/transcript`, options);
    }
    /**
     * Returns details of all sessions of an App.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.sessions.getSessions('app_id', {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    getSessions(appId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/sessions`, { query, ...options });
    }
}
