// cloudflare/resources/realtime-kit/webhooks.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
export class Webhooks extends APIResource {
    /**
     * Adds a new webhook to an App.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.createWebhook(
     *     'app_id',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       events: [
     *         'meeting.started',
     *         'meeting.ended',
     *         'meeting.participantJoined',
     *         'meeting.participantLeft',
     *         'meeting.chatSynced',
     *         'recording.statusUpdate',
     *         'livestreaming.statusUpdate',
     *         'meeting.transcript',
     *         'meeting.summary',
     *       ],
     *       name: 'All events webhook',
     *       url: 'https://webhook.site/b23a5bbd-c7b0-4ced-a9e2-78ae7889897e',
     *     },
     *   );
     * ```
     */
    createWebhook(appId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/realtime/kit/${appId}/webhooks`, { body, ...options });
    }
    /**
     * Removes a webhook for the given webhook ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.deleteWebhook(
     *     'app_id',
     *     'webhook_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    deleteWebhook(appId, webhookId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/realtime/kit/${appId}/webhooks/${webhookId}`, options);
    }
    /**
     * Edits the webhook details for the given webhook ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.editWebhook(
     *     'app_id',
     *     'webhook_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    editWebhook(appId, webhookId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/realtime/kit/${appId}/webhooks/${webhookId}`, {
            body,
            ...options,
        });
    }
    /**
     * Returns webhook details for the given webhook ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.getWebhookById(
     *     'app_id',
     *     'webhook_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    getWebhookById(appId, webhookId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/webhooks/${webhookId}`, options);
    }
    /**
     * Returns details of all webhooks for an App.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.getWebhooks('app_id', {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    getWebhooks(appId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/realtime/kit/${appId}/webhooks`, options);
    }
    /**
     * Replace all details for the given webhook ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.realtimeKit.webhooks.replaceWebhook(
     *     'app_id',
     *     'webhook_id',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       events: [
     *         'meeting.started',
     *         'meeting.ended',
     *         'meeting.participantJoined',
     *         'meeting.participantLeft',
     *         'meeting.chatSynced',
     *         'recording.statusUpdate',
     *         'livestreaming.statusUpdate',
     *         'meeting.transcript',
     *         'meeting.summary',
     *       ],
     *       name: 'All events webhook',
     *       url: 'https://webhook.site/b23a5bbd-c7b0-4ced-a9e2-78ae7889897e',
     *     },
     *   );
     * ```
     */
    replaceWebhook(appId, webhookId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/realtime/kit/${appId}/webhooks/${webhookId}`, {
            body,
            ...options,
        });
    }
}
