// cloudflare/resources/zero-trust/gateway/pacfiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PacfileListResponsesSinglePage = exports.Pacfiles = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class Pacfiles extends resource_1.APIResource {
    /**
     * Create a new Zero Trust Gateway PAC file.
     *
     * @example
     * ```ts
     * const pacfile =
     *   await client.zeroTrust.gateway.pacfiles.create({
     *     account_id: '699d98642c564d2e855e9661899b7252',
     *     contents:
     *       'function FindProxyForURL(url, host) { return "DIRECT"; }',
     *     name: 'Devops team',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/gateway/pacfiles`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update a configured Zero Trust Gateway PAC file.
     *
     * @example
     * ```ts
     * const pacfile =
     *   await client.zeroTrust.gateway.pacfiles.update(
     *     'ed35569b41ce4d1facfe683550f54086',
     *     {
     *       account_id: '699d98642c564d2e855e9661899b7252',
     *       contents:
     *         'function FindProxyForURL(url, host) { return "DIRECT"; }',
     *       description: 'PAC file for Devops team',
     *       name: 'Devops team',
     *     },
     *   );
     * ```
     */
    update(pacfileId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/gateway/pacfiles/${pacfileId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all Zero Trust Gateway PAC files for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const pacfileListResponse of client.zeroTrust.gateway.pacfiles.list(
     *   { account_id: '699d98642c564d2e855e9661899b7252' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/gateway/pacfiles`, PacfileListResponsesSinglePage, options);
    }
    /**
     * Delete a configured Zero Trust Gateway PAC file.
     *
     * @example
     * ```ts
     * const pacfile =
     *   await client.zeroTrust.gateway.pacfiles.delete(
     *     'ed35569b41ce4d1facfe683550f54086',
     *     { account_id: '699d98642c564d2e855e9661899b7252' },
     *   );
     * ```
     */
    delete(pacfileId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/gateway/pacfiles/${pacfileId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a single Zero Trust Gateway PAC file.
     *
     * @example
     * ```ts
     * const pacfile = await client.zeroTrust.gateway.pacfiles.get(
     *   'ed35569b41ce4d1facfe683550f54086',
     *   { account_id: '699d98642c564d2e855e9661899b7252' },
     * );
     * ```
     */
    get(pacfileId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/gateway/pacfiles/${pacfileId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Pacfiles = Pacfiles;
class PacfileListResponsesSinglePage extends pagination_1.SinglePage {
}
exports.PacfileListResponsesSinglePage = PacfileListResponsesSinglePage;
Pacfiles.PacfileListResponsesSinglePage = PacfileListResponsesSinglePage;
