// cloudflare/resources/api-gateway/labels/user/user.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as ResourcesAPI from "./resources/resources.mjs";
import { Resources } from "./resources/resources.mjs";
import { SinglePage } from "../../../../pagination.mjs";
export class User extends APIResource {
    constructor() {
        super(...arguments);
        this.resources = new ResourcesAPI.Resources(this._client);
    }
    /**
     * Update all fields on a label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.update(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    update(name, params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Delete user label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.delete(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(name, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/api_gateway/labels/user/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create user labels
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBulkCreateResponse of client.apiGateway.labels.user.bulkCreate(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [{ name: 'login' }],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkCreate(params, options) {
        const { zone_id, body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/labels/user`, UserBulkCreateResponsesSinglePage, { body: body, method: 'post', ...options });
    }
    /**
     * Delete user labels
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBulkDeleteResponse of client.apiGateway.labels.user.bulkDelete(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkDelete(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/labels/user`, UserBulkDeleteResponsesSinglePage, { method: 'delete', ...options });
    }
    /**
     * Update certain fields on a label
     *
     * @example
     * ```ts
     * const response = await client.apiGateway.labels.user.edit(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    edit(name, params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve user label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.get(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(name, params, options) {
        const { zone_id, ...query } = params;
        return this._client.get(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
export class UserBulkCreateResponsesSinglePage extends SinglePage {
}
export class UserBulkDeleteResponsesSinglePage extends SinglePage {
}
User.UserBulkCreateResponsesSinglePage = UserBulkCreateResponsesSinglePage;
User.UserBulkDeleteResponsesSinglePage = UserBulkDeleteResponsesSinglePage;
User.Resources = Resources;
