// cloudflare/resources/api-gateway/labels/user/user.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserBulkDeleteResponsesSinglePage = exports.UserBulkCreateResponsesSinglePage = exports.User = void 0;
const resource_1 = require("../../../../resource.js");
const ResourcesAPI = __importStar(require("./resources/resources.js"));
const resources_1 = require("./resources/resources.js");
const pagination_1 = require("../../../../pagination.js");
class User extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.resources = new ResourcesAPI.Resources(this._client);
    }
    /**
     * Update all fields on a label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.update(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    update(name, params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Delete user label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.delete(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(name, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/api_gateway/labels/user/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create user labels
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBulkCreateResponse of client.apiGateway.labels.user.bulkCreate(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [{ name: 'login' }],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkCreate(params, options) {
        const { zone_id, body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/labels/user`, UserBulkCreateResponsesSinglePage, { body: body, method: 'post', ...options });
    }
    /**
     * Delete user labels
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBulkDeleteResponse of client.apiGateway.labels.user.bulkDelete(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkDelete(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/api_gateway/labels/user`, UserBulkDeleteResponsesSinglePage, { method: 'delete', ...options });
    }
    /**
     * Update certain fields on a label
     *
     * @example
     * ```ts
     * const response = await client.apiGateway.labels.user.edit(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    edit(name, params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve user label
     *
     * @example
     * ```ts
     * const user = await client.apiGateway.labels.user.get(
     *   'login',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(name, params, options) {
        const { zone_id, ...query } = params;
        return this._client.get(`/zones/${zone_id}/api_gateway/labels/user/${name}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.User = User;
class UserBulkCreateResponsesSinglePage extends pagination_1.SinglePage {
}
exports.UserBulkCreateResponsesSinglePage = UserBulkCreateResponsesSinglePage;
class UserBulkDeleteResponsesSinglePage extends pagination_1.SinglePage {
}
exports.UserBulkDeleteResponsesSinglePage = UserBulkDeleteResponsesSinglePage;
User.UserBulkCreateResponsesSinglePage = UserBulkCreateResponsesSinglePage;
User.UserBulkDeleteResponsesSinglePage = UserBulkDeleteResponsesSinglePage;
User.Resources = resources_1.Resources;
