// cloudflare/resources/ddos-protection/advanced-tcp-protection/advanced-tcp-protection.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as StatusAPI from "./status.mjs";
import { Status } from "./status.mjs";
import * as AllowlistAPI from "./allowlist/allowlist.mjs";
import { Allowlist, AllowlistListResponsesV4PagePaginationArray, } from "./allowlist/allowlist.mjs";
import * as PrefixesAPI from "./prefixes/prefixes.mjs";
import { PrefixBulkCreateResponsesSinglePage, PrefixListResponsesV4PagePaginationArray, Prefixes, } from "./prefixes/prefixes.mjs";
import * as SynProtectionAPI from "./syn-protection/syn-protection.mjs";
import { SynProtection } from "./syn-protection/syn-protection.mjs";
import * as TCPFlowProtectionAPI from "./tcp-flow-protection/tcp-flow-protection.mjs";
import { TCPFlowProtection } from "./tcp-flow-protection/tcp-flow-protection.mjs";
export class AdvancedTCPProtection extends APIResource {
    constructor() {
        super(...arguments);
        this.allowlist = new AllowlistAPI.Allowlist(this._client);
        this.prefixes = new PrefixesAPI.Prefixes(this._client);
        this.synProtection = new SynProtectionAPI.SynProtection(this._client);
        this.tcpFlowProtection = new TCPFlowProtectionAPI.TCPFlowProtection(this._client);
        this.status = new StatusAPI.Status(this._client);
    }
}
AdvancedTCPProtection.Allowlist = Allowlist;
AdvancedTCPProtection.AllowlistListResponsesV4PagePaginationArray =
    AllowlistListResponsesV4PagePaginationArray;
AdvancedTCPProtection.Prefixes = Prefixes;
AdvancedTCPProtection.PrefixListResponsesV4PagePaginationArray = PrefixListResponsesV4PagePaginationArray;
AdvancedTCPProtection.PrefixBulkCreateResponsesSinglePage = PrefixBulkCreateResponsesSinglePage;
AdvancedTCPProtection.SynProtection = SynProtection;
AdvancedTCPProtection.TCPFlowProtection = TCPFlowProtection;
AdvancedTCPProtection.Status = Status;
