// cloudflare/resources/ddos-protection/advanced-tcp-protection/advanced-tcp-protection.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvancedTCPProtection = void 0;
const resource_1 = require("../../../resource.js");
const StatusAPI = __importStar(require("./status.js"));
const status_1 = require("./status.js");
const AllowlistAPI = __importStar(require("./allowlist/allowlist.js"));
const allowlist_1 = require("./allowlist/allowlist.js");
const PrefixesAPI = __importStar(require("./prefixes/prefixes.js"));
const prefixes_1 = require("./prefixes/prefixes.js");
const SynProtectionAPI = __importStar(require("./syn-protection/syn-protection.js"));
const syn_protection_1 = require("./syn-protection/syn-protection.js");
const TCPFlowProtectionAPI = __importStar(require("./tcp-flow-protection/tcp-flow-protection.js"));
const tcp_flow_protection_1 = require("./tcp-flow-protection/tcp-flow-protection.js");
class AdvancedTCPProtection extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.allowlist = new AllowlistAPI.Allowlist(this._client);
        this.prefixes = new PrefixesAPI.Prefixes(this._client);
        this.synProtection = new SynProtectionAPI.SynProtection(this._client);
        this.tcpFlowProtection = new TCPFlowProtectionAPI.TCPFlowProtection(this._client);
        this.status = new StatusAPI.Status(this._client);
    }
}
exports.AdvancedTCPProtection = AdvancedTCPProtection;
AdvancedTCPProtection.Allowlist = allowlist_1.Allowlist;
AdvancedTCPProtection.AllowlistListResponsesV4PagePaginationArray =
    allowlist_1.AllowlistListResponsesV4PagePaginationArray;
AdvancedTCPProtection.Prefixes = prefixes_1.Prefixes;
AdvancedTCPProtection.PrefixListResponsesV4PagePaginationArray = prefixes_1.PrefixListResponsesV4PagePaginationArray;
AdvancedTCPProtection.PrefixBulkCreateResponsesSinglePage = prefixes_1.PrefixBulkCreateResponsesSinglePage;
AdvancedTCPProtection.SynProtection = syn_protection_1.SynProtection;
AdvancedTCPProtection.TCPFlowProtection = tcp_flow_protection_1.TCPFlowProtection;
AdvancedTCPProtection.Status = status_1.Status;
