// cloudflare/resources/ddos-protection/advanced-tcp-protection/tcp-flow-protection/filters/filters.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../../resource.mjs";
import * as ItemsAPI from "./items.mjs";
import { Items, } from "./items.mjs";
import { V4PagePaginationArray } from "../../../../../pagination.mjs";
export class Filters extends APIResource {
    constructor() {
        super(...arguments);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a TCP Flow Protection filter for an account.
     *
     * @example
     * ```ts
     * const filter =
     *   await client.DDoSProtection.advancedTCPProtection.tcpFlowProtection.filters.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       expression:
     *         'ip.dst in { 192.0.2.0/24 198.51.100.0/24 } and tcp.srcport in { 80 443 10000..65535 }',
     *       mode: 'mode',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/tcp_flow_protection/filters`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all TCP Flow Protection filters for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const filterListResponse of client.DDoSProtection.advancedTCPProtection.tcpFlowProtection.filters.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/tcp_flow_protection/filters`, FilterListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete all TCP Flow Protection filters for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.DDoSProtection.advancedTCPProtection.tcpFlowProtection.filters.bulkDelete(
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkDelete(params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/tcp_flow_protection/filters`, options);
    }
}
export class FilterListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Filters.FilterListResponsesV4PagePaginationArray = FilterListResponsesV4PagePaginationArray;
Filters.Items = Items;
