// cloudflare/resources/cache/origin-cloud-regions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OriginCloudRegions = void 0;
const resource_1 = require("../../resource.js");
class OriginCloudRegions extends resource_1.APIResource {
    /**
     * Adds a single IP-to-cloud-region mapping for the zone. The IP must be a valid
     * IPv4 or IPv6 address and is normalized to canonical form before storage (RFC
     * 5952 for IPv6). Returns 400 (code 1145) if a mapping for that IP already exists
     * — use PATCH to update an existing entry. The vendor and region are validated
     * against the list from
     * `GET /zones/{zone_id}/cache/origin_cloud_regions/supported_regions`.
     *
     * @example
     * ```ts
     * const originCloudRegion =
     *   await client.cache.originCloudRegions.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     ip: '192.0.2.1',
     *     region: 'us-east-1',
     *     vendor: 'aws',
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/cache/origin_cloud_regions`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns all IP-to-cloud-region mappings configured for the zone. Each mapping
     * tells Cloudflare which cloud vendor and region hosts the origin at that IP,
     * enabling the edge to route via the nearest Tiered Cache upper-tier co-located
     * with that cloud provider. Returns an empty array when no mappings exist.
     *
     * @example
     * ```ts
     * const originCloudRegions =
     *   await client.cache.originCloudRegions.list({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    list(params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/cache/origin_cloud_regions`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Removes the cloud region mapping for a single origin IP address. The IP path
     * parameter is normalized before lookup. Returns the deleted entry on success.
     * Returns 404 (code 1163) if no mapping exists for the specified IP. When the last
     * mapping for the zone is removed the underlying rule record is also deleted.
     *
     * @example
     * ```ts
     * const originCloudRegion =
     *   await client.cache.originCloudRegions.delete(
     *     '192.0.2.1',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(originIP, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/cache/origin_cloud_regions/${originIP}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Removes up to 100 IP-to-cloud-region mappings in a single request. Each IP is
     * validated independently — successfully deleted items are returned in the
     * `succeeded` array and IPs that could not be found or are invalid are returned in
     * the `failed` array.
     *
     * @example
     * ```ts
     * const response =
     *   await client.cache.originCloudRegions.bulkDelete({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    bulkDelete(params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/cache/origin_cloud_regions/batch`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Adds or updates up to 100 IP-to-cloud-region mappings in a single request. Each
     * item is validated independently — valid items are applied and invalid items are
     * returned in the `failed` array. The vendor and region for every item are
     * validated against the list from
     * `GET /zones/{zone_id}/cache/origin_cloud_regions/supported_regions`.
     *
     * @example
     * ```ts
     * const response =
     *   await client.cache.originCloudRegions.bulkEdit({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [
     *       {
     *         ip: '192.0.2.1',
     *         region: 'us-east-1',
     *         vendor: 'aws',
     *       },
     *       {
     *         ip: '2001:db8::1',
     *         region: 'us-central1',
     *         vendor: 'gcp',
     *       },
     *     ],
     *   });
     * ```
     */
    bulkEdit(params, options) {
        const { zone_id, body } = params;
        return this._client.patch(`/zones/${zone_id}/cache/origin_cloud_regions/batch`, {
            body: body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Adds or updates a single IP-to-cloud-region mapping for the zone. Unlike POST,
     * this operation is idempotent — if a mapping for the IP already exists it is
     * overwritten. Returns the complete updated list of all mappings for the zone.
     * Returns 403 (code 1164) when the zone has reached the limit of 3,500 IP
     * mappings.
     *
     * @example
     * ```ts
     * const response = await client.cache.originCloudRegions.edit(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     ip: '2001:db8::1',
     *     region: 'us-central1',
     *     vendor: 'gcp',
     *   },
     * );
     * ```
     */
    edit(params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/cache/origin_cloud_regions`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns the cloud region mapping for a single origin IP address. The IP path
     * parameter is normalized before lookup (RFC 5952 for IPv6). Returns 404
     * (code 1142) if the zone has no mappings or if the specified IP has no mapping.
     *
     * @example
     * ```ts
     * const originCloudRegion =
     *   await client.cache.originCloudRegions.get('192.0.2.1', {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    get(originIP, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/cache/origin_cloud_regions/${originIP}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns the cloud vendors and regions that are valid values for origin cloud
     * region mappings. Each region includes the Tiered Cache upper-tier colocation
     * codes that will be used for cache routing when a mapping targeting that region
     * is active. Requires the zone to have Tiered Cache enabled.
     *
     * @example
     * ```ts
     * const response =
     *   await client.cache.originCloudRegions.supportedRegions({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   });
     * ```
     */
    supportedRegions(params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/cache/origin_cloud_regions/supported_regions`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.OriginCloudRegions = OriginCloudRegions;
