// cloudflare/resources/iam/sso.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import { SinglePage } from "../../pagination.mjs";
export class SSO extends APIResource {
    /**
     * Initialize new SSO connector
     *
     * @example
     * ```ts
     * const sso = await client.iam.sso.create({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   email_domain: 'example.com',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/sso_connectors`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update SSO connector state
     *
     * @example
     * ```ts
     * const sso = await client.iam.sso.update(
     *   '023e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    update(ssoConnectorId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/sso_connectors/${ssoConnectorId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get all SSO connectors
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ssoListResponse of client.iam.sso.list({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     * })) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/sso_connectors`, SSOListResponsesSinglePage, options);
    }
    /**
     * Delete SSO connector
     *
     * @example
     * ```ts
     * const sso = await client.iam.sso.delete(
     *   '023e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(ssoConnectorId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/sso_connectors/${ssoConnectorId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Begin SSO connector verification
     *
     * @example
     * ```ts
     * const response = await client.iam.sso.beginVerification(
     *   '023e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    beginVerification(ssoConnectorId, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/sso_connectors/${ssoConnectorId}/begin_verification`, options);
    }
    /**
     * Get single SSO connector
     *
     * @example
     * ```ts
     * const sso = await client.iam.sso.get(
     *   '023e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(ssoConnectorId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/sso_connectors/${ssoConnectorId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class SSOListResponsesSinglePage extends SinglePage {
}
SSO.SSOListResponsesSinglePage = SSOListResponsesSinglePage;
