// cloudflare/resources/brand-protection/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2 = exports.Queries = exports.Matches = exports.Logos = exports.LogoMatches = exports.BrandProtection = void 0;
var brand_protection_1 = require("./brand-protection.js");
Object.defineProperty(exports, "BrandProtection", { enumerable: true, get: function () { return brand_protection_1.BrandProtection; } });
var logo_matches_1 = require("./logo-matches.js");
Object.defineProperty(exports, "LogoMatches", { enumerable: true, get: function () { return logo_matches_1.LogoMatches; } });
var logos_1 = require("./logos.js");
Object.defineProperty(exports, "Logos", { enumerable: true, get: function () { return logos_1.Logos; } });
var matches_1 = require("./matches.js");
Object.defineProperty(exports, "Matches", { enumerable: true, get: function () { return matches_1.Matches; } });
var queries_1 = require("./queries.js");
Object.defineProperty(exports, "Queries", { enumerable: true, get: function () { return queries_1.Queries; } });
var index_1 = require("./v2/index.js");
Object.defineProperty(exports, "V2", { enumerable: true, get: function () { return index_1.V2; } });
