// cloudflare/resources/token-validation/rules.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenValidationRulesSinglePage = exports.TokenValidationRulesV4PagePaginationArray = exports.Rules = void 0;
const resource_1 = require("../../resource.js");
const pagination_1 = require("../../pagination.js");
class Rules extends resource_1.APIResource {
    /**
     * Create a token validation rule.
     *
     * @example
     * ```ts
     * const tokenValidationRule =
     *   await client.tokenValidation.rules.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     action: 'log',
     *     description:
     *       'Long description for Token Validation Rule',
     *     enabled: true,
     *     expression:
     *       'is_jwt_valid("52973293-cb04-4a97-8f55-e7d2ad1107dd") or is_jwt_valid("46eab8d1-6376-45e3-968f-2c649d77d423")',
     *     selector: {},
     *     title: 'Example Token Validation Rule',
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/token_validation/rules`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List token validation rules
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const tokenValidationRule of client.tokenValidation.rules.list(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { zone_id, ...query } = params;
        return this._client.getAPIList(`/zones/${zone_id}/token_validation/rules`, TokenValidationRulesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete a zone token validation rule.
     *
     * @example
     * ```ts
     * const rule = await client.tokenValidation.rules.delete(
     *   '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(ruleId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/token_validation/rules/${ruleId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create zone token validation rules.
     *
     * A request can create multiple Token Validation Rules.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const tokenValidationRule of client.tokenValidation.rules.bulkCreate(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [
     *       {
     *         action: 'log',
     *         description:
     *           'Long description for Token Validation Rule',
     *         enabled: true,
     *         expression:
     *           'is_jwt_valid("52973293-cb04-4a97-8f55-e7d2ad1107dd") or is_jwt_valid("46eab8d1-6376-45e3-968f-2c649d77d423")',
     *         selector: {},
     *         title: 'Example Token Validation Rule',
     *       },
     *     ],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkCreate(params, options) {
        const { zone_id, body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/token_validation/rules/bulk`, TokenValidationRulesSinglePage, { body: body, method: 'post', ...options });
    }
    /**
     * Edit token validation rules.
     *
     * A request can update multiple Token Validation Rules.
     *
     * Rules can be re-ordered using the `position` field.
     *
     * Returns all updated rules.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const tokenValidationRule of client.tokenValidation.rules.bulkEdit(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [{ id: '0d9bf70c-92e1-4bb3-9411-34a3bcc59003' }],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkEdit(params, options) {
        const { zone_id, body } = params;
        return this._client.getAPIList(`/zones/${zone_id}/token_validation/rules/bulk`, TokenValidationRulesSinglePage, { body: body, method: 'patch', ...options });
    }
    /**
     * Edit a zone token validation rule.
     *
     * @example
     * ```ts
     * const tokenValidationRule =
     *   await client.tokenValidation.rules.edit(
     *     '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    edit(ruleId, params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/token_validation/rules/${ruleId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a zone token validation rule.
     *
     * @example
     * ```ts
     * const tokenValidationRule =
     *   await client.tokenValidation.rules.get(
     *     '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(ruleId, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/token_validation/rules/${ruleId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Rules = Rules;
class TokenValidationRulesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.TokenValidationRulesV4PagePaginationArray = TokenValidationRulesV4PagePaginationArray;
class TokenValidationRulesSinglePage extends pagination_1.SinglePage {
}
exports.TokenValidationRulesSinglePage = TokenValidationRulesSinglePage;
Rules.TokenValidationRulesV4PagePaginationArray = TokenValidationRulesV4PagePaginationArray;
Rules.TokenValidationRulesSinglePage = TokenValidationRulesSinglePage;
