// cloudflare/resources/intel/attack-surface-report/issues.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { V4PagePagination } from "../../../pagination.mjs";
export class Issues extends APIResource {
    /**
     * Lists all Security Center issues for the account, showing active security
     * problems requiring attention.
     *
     * @deprecated
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/intel/attack-surface-report/issues`, IssueListResponsesV4PagePagination, { query, ...options });
    }
    /**
     * Retrieves Security Center issue counts aggregated by classification class.
     *
     * @deprecated
     */
    class(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/intel/attack-surface-report/issues/class`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Deprecated endpoint for archiving Security Center insights. Use the newer
     * archive-security-center-insight endpoint instead.
     *
     * @deprecated
     */
    dismiss(issueId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/intel/attack-surface-report/${issueId}/dismiss`, {
            body,
            ...options,
        });
    }
    /**
     * Retrieves Security Center issue counts aggregated by severity level.
     *
     * @deprecated
     */
    severity(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/intel/attack-surface-report/issues/severity`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves Security Center issue counts aggregated by issue type.
     *
     * @deprecated
     */
    type(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/intel/attack-surface-report/issues/type`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
export class IssueListResponsesV4PagePagination extends V4PagePagination {
}
Issues.IssueListResponsesV4PagePagination = IssueListResponsesV4PagePagination;
