// cloudflare/resources/alerting/silences.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import { SinglePage } from "../../pagination.mjs";
export class Silences extends APIResource {
    /**
     * Creates a new silence for an account.
     *
     * @example
     * ```ts
     * const silence = await client.alerting.silences.create({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   body: [{}],
     * });
     * ```
     */
    create(params, options) {
        const { account_id, body } = params;
        return this._client.post(`/accounts/${account_id}/alerting/v3/silences`, { body: body, ...options });
    }
    /**
     * Updates existing silences for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const silenceUpdateResponse of client.alerting.silences.update(
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [{}],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    update(params, options) {
        const { account_id, body } = params;
        return this._client.getAPIList(`/accounts/${account_id}/alerting/v3/silences`, SilenceUpdateResponsesSinglePage, { body: body, method: 'put', ...options });
    }
    /**
     * Gets a list of silences for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const silenceListResponse of client.alerting.silences.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/alerting/v3/silences`, SilenceListResponsesSinglePage, options);
    }
    /**
     * Deletes an existing silence for an account.
     *
     * @example
     * ```ts
     * const silence = await client.alerting.silences.delete(
     *   'f878e90c23f44126ae3cfc399f646977',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(silenceId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/alerting/v3/silences/${silenceId}`, options);
    }
    /**
     * Gets a specific silence for an account.
     *
     * @example
     * ```ts
     * const silence = await client.alerting.silences.get(
     *   'f878e90c23f44126ae3cfc399f646977',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(silenceId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/alerting/v3/silences/${silenceId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class SilenceUpdateResponsesSinglePage extends SinglePage {
}
export class SilenceListResponsesSinglePage extends SinglePage {
}
Silences.SilenceUpdateResponsesSinglePage = SilenceUpdateResponsesSinglePage;
Silences.SilenceListResponsesSinglePage = SilenceListResponsesSinglePage;
