// cloudflare/resources/radar/email/security/security.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import { isRequestOptions } from "../../../../core.mjs";
import * as SummaryAPI from "./summary.mjs";
import { Summary, } from "./summary.mjs";
import * as TimeseriesGroupsAPI from "./timeseries-groups.mjs";
import { TimeseriesGroups, } from "./timeseries-groups.mjs";
import * as TopAPI from "./top/top.mjs";
import { Top } from "./top/top.mjs";
export class Security extends APIResource {
    constructor() {
        super(...arguments);
        this.top = new TopAPI.Top(this._client);
        this.summary = new SummaryAPI.Summary(this._client);
        this.timeseriesGroups = new TimeseriesGroupsAPI.TimeseriesGroups(this._client);
    }
    summaryV2(dimension, query = {}, options) {
        if (isRequestOptions(query)) {
            return this.summaryV2(dimension, {}, query);
        }
        return this._client.get(`/radar/email/security/summary/${dimension}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    timeseriesGroupsV2(dimension, query = {}, options) {
        if (isRequestOptions(query)) {
            return this.timeseriesGroupsV2(dimension, {}, query);
        }
        return this._client.get(`/radar/email/security/timeseries_groups/${dimension}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
Security.Top = Top;
Security.Summary = Summary;
Security.TimeseriesGroups = TimeseriesGroups;
