// cloudflare/resources/ddos-protection/advanced-tcp-protection/syn-protection/rules/rules.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../../resource.mjs";
import * as ItemsAPI from "./items.mjs";
import { Items, } from "./items.mjs";
import { V4PagePaginationArray } from "../../../../../pagination.mjs";
export class Rules extends APIResource {
    constructor() {
        super(...arguments);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a SYN Protection rule for an account.
     *
     * @example
     * ```ts
     * const rule =
     *   await client.DDoSProtection.advancedTCPProtection.synProtection.rules.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       burst_sensitivity: 'burst_sensitivity',
     *       mode: 'mode',
     *       name: 'name',
     *       rate_sensitivity: 'rate_sensitivity',
     *       scope: 'scope',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/rules`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all SYN Protection rules for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ruleListResponse of client.DDoSProtection.advancedTCPProtection.synProtection.rules.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/rules`, RuleListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete all SYN Protection rules for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.DDoSProtection.advancedTCPProtection.synProtection.rules.bulkDelete(
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkDelete(params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/syn_protection/rules`, options);
    }
}
export class RuleListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Rules.RuleListResponsesV4PagePaginationArray = RuleListResponsesV4PagePaginationArray;
Rules.Items = Items;
