// cloudflare/resources/pipelines/pipelines.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelineListV1ResponsesV4PagePaginationArray = exports.Pipelines = void 0;
const resource_1 = require("../../resource.js");
const SinksAPI = __importStar(require("./sinks.js"));
const sinks_1 = require("./sinks.js");
const StreamsAPI = __importStar(require("./streams.js"));
const streams_1 = require("./streams.js");
const pagination_1 = require("../../pagination.js");
class Pipelines extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.sinks = new SinksAPI.Sinks(this._client);
        this.streams = new StreamsAPI.Streams(this._client);
    }
    /**
     * [DEPRECATED] Create a new pipeline. Use the new /pipelines/v1/pipelines endpoint
     * instead.
     *
     * @deprecated Use create_v1 instead. This endpoint will be removed in the future.
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] Update an existing pipeline. Use the new /pipelines/v1/pipelines
     * endpoint instead.
     *
     * @deprecated The v1 API does not support updates. This endpoint will be removed in the future.
     */
    update(pipelineName, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/pipelines/${pipelineName}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] List, filter, and paginate pipelines in an account. Use the new
     * /pipelines/v1/pipelines endpoint instead.
     *
     * @deprecated Use list_v1 instead. This endpoint will be removed in the future.
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/pipelines`, { query, ...options });
    }
    /**
     * [DEPRECATED] Delete a pipeline. Use the new /pipelines/v1/pipelines endpoint
     * instead.
     *
     * @deprecated Use delete_v1 instead. This endpoint will be removed in the future.
     */
    delete(pipelineName, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/${pipelineName}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Create a new Pipeline.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.createV1({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   name: 'my_pipeline',
     *   sql: 'insert into sink select * from source;',
     * });
     * ```
     */
    createV1(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/pipelines`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Delete Pipeline in Account.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.deleteV1(
     *   '043e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    deleteV1(pipelineId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/v1/pipelines/${pipelineId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] Get configuration of a pipeline. Use the new
     * /pipelines/v1/pipelines endpoint instead.
     *
     * @deprecated Use get_v1 instead. This endpoint will be removed in the future.
     */
    get(pipelineName, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/${pipelineName}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get Pipelines Details.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.getV1(
     *   '043e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    getV1(pipelineId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/v1/pipelines/${pipelineId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * List/Filter Pipelines in Account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const pipelineListV1Response of client.pipelines.listV1(
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    listV1(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/pipelines/v1/pipelines`, PipelineListV1ResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Validate Arroyo SQL.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.validateSql({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   sql: 'insert into sink select * from source;',
     * });
     * ```
     */
    validateSql(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/validate_sql`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Pipelines = Pipelines;
class PipelineListV1ResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.PipelineListV1ResponsesV4PagePaginationArray = PipelineListV1ResponsesV4PagePaginationArray;
Pipelines.PipelineListV1ResponsesV4PagePaginationArray = PipelineListV1ResponsesV4PagePaginationArray;
Pipelines.Sinks = sinks_1.Sinks;
Pipelines.SinkListResponsesV4PagePaginationArray = sinks_1.SinkListResponsesV4PagePaginationArray;
Pipelines.Streams = streams_1.Streams;
Pipelines.StreamListResponsesV4PagePaginationArray = streams_1.StreamListResponsesV4PagePaginationArray;
