// cloudflare/resources/pipelines/pipelines.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as SinksAPI from "./sinks.mjs";
import { SinkListResponsesV4PagePaginationArray, Sinks, } from "./sinks.mjs";
import * as StreamsAPI from "./streams.mjs";
import { StreamListResponsesV4PagePaginationArray, Streams, } from "./streams.mjs";
import { V4PagePaginationArray } from "../../pagination.mjs";
export class Pipelines extends APIResource {
    constructor() {
        super(...arguments);
        this.sinks = new SinksAPI.Sinks(this._client);
        this.streams = new StreamsAPI.Streams(this._client);
    }
    /**
     * [DEPRECATED] Create a new pipeline. Use the new /pipelines/v1/pipelines endpoint
     * instead.
     *
     * @deprecated Use create_v1 instead. This endpoint will be removed in the future.
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] Update an existing pipeline. Use the new /pipelines/v1/pipelines
     * endpoint instead.
     *
     * @deprecated The v1 API does not support updates. This endpoint will be removed in the future.
     */
    update(pipelineName, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/pipelines/${pipelineName}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] List, filter, and paginate pipelines in an account. Use the new
     * /pipelines/v1/pipelines endpoint instead.
     *
     * @deprecated Use list_v1 instead. This endpoint will be removed in the future.
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/pipelines`, { query, ...options });
    }
    /**
     * [DEPRECATED] Delete a pipeline. Use the new /pipelines/v1/pipelines endpoint
     * instead.
     *
     * @deprecated Use delete_v1 instead. This endpoint will be removed in the future.
     */
    delete(pipelineName, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/${pipelineName}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Create a new Pipeline.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.createV1({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   name: 'my_pipeline',
     *   sql: 'insert into sink select * from source;',
     * });
     * ```
     */
    createV1(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/pipelines`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Delete Pipeline in Account.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.deleteV1(
     *   '043e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    deleteV1(pipelineId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/v1/pipelines/${pipelineId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * [DEPRECATED] Get configuration of a pipeline. Use the new
     * /pipelines/v1/pipelines endpoint instead.
     *
     * @deprecated Use get_v1 instead. This endpoint will be removed in the future.
     */
    get(pipelineName, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/${pipelineName}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get Pipelines Details.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.getV1(
     *   '043e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    getV1(pipelineId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/v1/pipelines/${pipelineId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * List/Filter Pipelines in Account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const pipelineListV1Response of client.pipelines.listV1(
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    listV1(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/pipelines/v1/pipelines`, PipelineListV1ResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Validate Arroyo SQL.
     *
     * @example
     * ```ts
     * const response = await client.pipelines.validateSql({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   sql: 'insert into sink select * from source;',
     * });
     * ```
     */
    validateSql(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/validate_sql`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
export class PipelineListV1ResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Pipelines.PipelineListV1ResponsesV4PagePaginationArray = PipelineListV1ResponsesV4PagePaginationArray;
Pipelines.Sinks = Sinks;
Pipelines.SinkListResponsesV4PagePaginationArray = SinkListResponsesV4PagePaginationArray;
Pipelines.Streams = Streams;
Pipelines.StreamListResponsesV4PagePaginationArray = StreamListResponsesV4PagePaginationArray;
