// cloudflare/resources/pipelines/sinks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SinkListResponsesV4PagePaginationArray = exports.Sinks = void 0;
const resource_1 = require("../../resource.js");
const pagination_1 = require("../../pagination.js");
class Sinks extends resource_1.APIResource {
    /**
     * Create a new Sink.
     *
     * @example
     * ```ts
     * const sink = await client.pipelines.sinks.create({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   name: 'my_sink',
     *   type: 'r2',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/sinks`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List/Filter Sinks in Account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const sinkListResponse of client.pipelines.sinks.list(
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/pipelines/v1/sinks`, SinkListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete Pipeline in Account.
     *
     * @example
     * ```ts
     * const sink = await client.pipelines.sinks.delete(
     *   '0223105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(sinkId, params, options) {
        const { account_id, force } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/v1/sinks/${sinkId}`, {
            query: { force },
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get Sink Details.
     *
     * @example
     * ```ts
     * const sink = await client.pipelines.sinks.get(
     *   '0223105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(sinkId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/v1/sinks/${sinkId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Sinks = Sinks;
class SinkListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.SinkListResponsesV4PagePaginationArray = SinkListResponsesV4PagePaginationArray;
Sinks.SinkListResponsesV4PagePaginationArray = SinkListResponsesV4PagePaginationArray;
