// cloudflare/resources/pipelines/streams.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import { V4PagePaginationArray } from "../../pagination.mjs";
export class Streams extends APIResource {
    /**
     * Create a new Stream.
     *
     * @example
     * ```ts
     * const stream = await client.pipelines.streams.create({
     *   account_id: '0123105f4ecef8ad9ca31a8372d0c353',
     *   name: 'my_stream',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/pipelines/v1/streams`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update a Stream.
     *
     * @example
     * ```ts
     * const stream = await client.pipelines.streams.update(
     *   '033e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    update(streamId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/pipelines/v1/streams/${streamId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List/Filter Streams in Account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const streamListResponse of client.pipelines.streams.list(
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/pipelines/v1/streams`, StreamListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete Stream in Account.
     *
     * @example
     * ```ts
     * const stream = await client.pipelines.streams.delete(
     *   '033e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(streamId, params, options) {
        const { account_id, force } = params;
        return this._client.delete(`/accounts/${account_id}/pipelines/v1/streams/${streamId}`, {
            query: { force },
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get Stream Details.
     *
     * @example
     * ```ts
     * const stream = await client.pipelines.streams.get(
     *   '033e105f4ecef8ad9ca31a8372d0c353',
     *   { account_id: '0123105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(streamId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/pipelines/v1/streams/${streamId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class StreamListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Streams.StreamListResponsesV4PagePaginationArray = StreamListResponsesV4PagePaginationArray;
