// cloudflare/resources/ai-gateway/billing/billing.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Billing = void 0;
const resource_1 = require("../../../resource.js");
const SpendingLimitAPI = __importStar(require("./spending-limit.js"));
const spending_limit_1 = require("./spending-limit.js");
const TopupAPI = __importStar(require("./topup/topup.js"));
const topup_1 = require("./topup/topup.js");
class Billing extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.topup = new TopupAPI.Topup(this._client);
        this.spendingLimit = new SpendingLimitAPI.SpendingLimit(this._client);
    }
    /**
     * Retrieve the current credit balance, payment method info, and top-up
     * configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.billing.creditBalance({
     *     account_id: 'account_id',
     *   });
     * ```
     */
    creditBalance(params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/billing/credit-balance`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve a list of past invoices with pagination, optionally filtered by type.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.billing.invoiceHistory({
     *     account_id: 'account_id',
     *   });
     * ```
     */
    invoiceHistory(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/billing/invoice-history`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve a preview of the upcoming invoice including line items and tax.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.billing.invoicePreview({
     *     account_id: 'account_id',
     *   });
     * ```
     */
    invoicePreview(params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/billing/invoice-preview`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve aggregated usage meter event summaries for the given time range.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.billing.usageHistory({
     *     account_id: 'account_id',
     *     value_grouping_window: 'day',
     *   });
     * ```
     */
    usageHistory(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/billing/usage-history`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Billing = Billing;
Billing.Topup = topup_1.Topup;
Billing.SpendingLimit = spending_limit_1.SpendingLimit;
