// cloudflare/resources/ddos-protection/advanced-tcp-protection/prefixes/prefixes.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as ItemsAPI from "./items.mjs";
import { Items, } from "./items.mjs";
import { SinglePage, V4PagePaginationArray } from "../../../../pagination.mjs";
export class Prefixes extends APIResource {
    constructor() {
        super(...arguments);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create a prefix for an account.
     *
     * @example
     * ```ts
     * const prefix =
     *   await client.DDoSProtection.advancedTCPProtection.prefixes.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       comment: 'comment',
     *       excluded: true,
     *       prefix: '192.0.2.0/24',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/prefixes`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all prefixes for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const prefixListResponse of client.DDoSProtection.advancedTCPProtection.prefixes.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/prefixes`, PrefixListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Create multiple prefixes for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const prefixBulkCreateResponse of client.DDoSProtection.advancedTCPProtection.prefixes.bulkCreate(
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     body: [
     *       {
     *         comment: 'comment',
     *         excluded: true,
     *         prefix: '192.0.2.0/24',
     *       },
     *     ],
     *   },
     * )) {
     *   // ...
     * }
     * ```
     */
    bulkCreate(params, options) {
        const { account_id, body } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/prefixes/bulk`, PrefixBulkCreateResponsesSinglePage, { body: body, method: 'post', ...options });
    }
    /**
     * Delete all prefixes for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.DDoSProtection.advancedTCPProtection.prefixes.bulkDelete(
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkDelete(params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/prefixes`, options);
    }
}
export class PrefixListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
export class PrefixBulkCreateResponsesSinglePage extends SinglePage {
}
Prefixes.PrefixListResponsesV4PagePaginationArray = PrefixListResponsesV4PagePaginationArray;
Prefixes.PrefixBulkCreateResponsesSinglePage = PrefixBulkCreateResponsesSinglePage;
Prefixes.Items = Items;
