// cloudflare/resources/load-balancers/monitor-groups/monitor-groups.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorGroupsSinglePage = exports.MonitorGroups = void 0;
const resource_1 = require("../../../resource.js");
const ReferencesAPI = __importStar(require("./references.js"));
const references_1 = require("./references.js");
const pagination_1 = require("../../../pagination.js");
class MonitorGroups extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.references = new ReferencesAPI.References(this._client);
    }
    /**
     * Create a new monitor group.
     *
     * @example
     * ```ts
     * const monitorGroup =
     *   await client.loadBalancers.monitorGroups.create({
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     description: 'Primary datacenter monitors',
     *     members: [
     *       {
     *         enabled: true,
     *         monitor_id: 'monitor_id',
     *         monitoring_only: false,
     *         must_be_healthy: true,
     *       },
     *     ],
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/load_balancers/monitor_groups`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Modify a configured monitor group.
     *
     * @example
     * ```ts
     * const monitorGroup =
     *   await client.loadBalancers.monitorGroups.update(
     *     '17b5962d775c646f3f9725cbc7a53df4',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       description: 'Primary datacenter monitors',
     *       members: [
     *         {
     *           enabled: true,
     *           monitor_id: 'monitor_id',
     *           monitoring_only: false,
     *           must_be_healthy: true,
     *         },
     *       ],
     *     },
     *   );
     * ```
     */
    update(monitorGroupId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/load_balancers/monitor_groups/${monitorGroupId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List configured monitor groups.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const monitorGroup of client.loadBalancers.monitorGroups.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/load_balancers/monitor_groups`, MonitorGroupsSinglePage, options);
    }
    /**
     * Delete a configured monitor group.
     *
     * @example
     * ```ts
     * const monitorGroup =
     *   await client.loadBalancers.monitorGroups.delete(
     *     '17b5962d775c646f3f9725cbc7a53df4',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(monitorGroupId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/load_balancers/monitor_groups/${monitorGroupId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Apply changes to an existing monitor group, overwriting the supplied properties.
     *
     * @example
     * ```ts
     * const monitorGroup =
     *   await client.loadBalancers.monitorGroups.edit(
     *     '17b5962d775c646f3f9725cbc7a53df4',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       description: 'Primary datacenter monitors',
     *       members: [
     *         {
     *           enabled: true,
     *           monitor_id: 'monitor_id',
     *           monitoring_only: false,
     *           must_be_healthy: true,
     *         },
     *       ],
     *     },
     *   );
     * ```
     */
    edit(monitorGroupId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/load_balancers/monitor_groups/${monitorGroupId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetch a single configured monitor group.
     *
     * @example
     * ```ts
     * const monitorGroup =
     *   await client.loadBalancers.monitorGroups.get(
     *     '17b5962d775c646f3f9725cbc7a53df4',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(monitorGroupId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/load_balancers/monitor_groups/${monitorGroupId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.MonitorGroups = MonitorGroups;
class MonitorGroupsSinglePage extends pagination_1.SinglePage {
}
exports.MonitorGroupsSinglePage = MonitorGroupsSinglePage;
MonitorGroups.MonitorGroupsSinglePage = MonitorGroupsSinglePage;
MonitorGroups.References = references_1.References;
MonitorGroups.ReferenceGetResponsesSinglePage = references_1.ReferenceGetResponsesSinglePage;
