// cloudflare/resources/origin-tls-client-auth/origin-tls-client-auth.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as HostnameCertificatesAPI from "./hostname-certificates.mjs";
import { HostnameCertificateListResponsesSinglePage, HostnameCertificates, } from "./hostname-certificates.mjs";
import * as HostnamesAPI from "./hostnames.mjs";
import { HostnameUpdateResponsesSinglePage, Hostnames, } from "./hostnames.mjs";
import * as SettingsAPI from "./settings.mjs";
import { Settings, } from "./settings.mjs";
import * as ZoneCertificatesAPI from "./zone-certificates.mjs";
import { ZoneCertificateListResponsesSinglePage, ZoneCertificates, } from "./zone-certificates.mjs";
import { SinglePage } from "../../pagination.mjs";
export class OriginTLSClientAuth extends APIResource {
    constructor() {
        super(...arguments);
        this.zoneCertificates = new ZoneCertificatesAPI.ZoneCertificates(this._client);
        this.hostnames = new HostnamesAPI.Hostnames(this._client);
        this.hostnameCertificates = new HostnameCertificatesAPI.HostnameCertificates(this._client);
        this.settings = new SettingsAPI.Settings(this._client);
    }
    /**
     * Upload your own certificate you want Cloudflare to use for edge-to-origin
     * communication to override the shared certificate. Please note that it is
     * important to keep only one certificate active. Also, make sure to enable
     * zone-level authenticated origin pulls by making a PUT call to settings endpoint
     * to see the uploaded certificate in use.
     *
     * @deprecated Use zone_certificates.create for zone-level certificates. This method will be removed in a future major version.
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/origin_tls_client_auth`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all client certificates configured for zone-level authenticated origin
     * pulls.
     *
     * @deprecated Use zone_certificates.list for zone-level certificates. This method will be removed in a future major version.
     */
    list(params, options) {
        const { zone_id } = params;
        return this._client.getAPIList(`/zones/${zone_id}/origin_tls_client_auth`, OriginTLSClientAuthListResponsesSinglePage, options);
    }
    /**
     * Removes a client certificate used for zone-level authenticated origin pulls.
     *
     * @deprecated Use zone_certificates.delete for zone-level certificates. This method will be removed in a future major version.
     */
    delete(certificateId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/origin_tls_client_auth/${certificateId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves details for a specific client certificate used in zone-level
     * authenticated origin pulls.
     *
     * @deprecated Use zone_certificates.get for zone-level certificates. This method will be removed in a future major version.
     */
    get(certificateId, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/origin_tls_client_auth/${certificateId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class OriginTLSClientAuthListResponsesSinglePage extends SinglePage {
}
OriginTLSClientAuth.OriginTLSClientAuthListResponsesSinglePage = OriginTLSClientAuthListResponsesSinglePage;
OriginTLSClientAuth.ZoneCertificates = ZoneCertificates;
OriginTLSClientAuth.ZoneCertificateListResponsesSinglePage = ZoneCertificateListResponsesSinglePage;
OriginTLSClientAuth.Hostnames = Hostnames;
OriginTLSClientAuth.HostnameUpdateResponsesSinglePage = HostnameUpdateResponsesSinglePage;
OriginTLSClientAuth.HostnameCertificates = HostnameCertificates;
OriginTLSClientAuth.HostnameCertificateListResponsesSinglePage = HostnameCertificateListResponsesSinglePage;
OriginTLSClientAuth.Settings = Settings;
