// cloudflare/resources/zones/environments.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Environments = void 0;
const resource_1 = require("../../resource.js");
class Environments extends resource_1.APIResource {
    /**
     * Create zone environments
     *
     * @example
     * ```ts
     * const environment = await client.zones.environments.create({
     *   zone_id: 'zone_id',
     *   environments: [
     *     {
     *       expression: 'expression',
     *       locked_on_deployment: true,
     *       name: 'name',
     *       position: {},
     *       ref: 'ref',
     *       version: 0,
     *     },
     *   ],
     * });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/environments`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Upsert zone environments
     *
     * @example
     * ```ts
     * const environment = await client.zones.environments.update({
     *   zone_id: 'zone_id',
     *   environments: [
     *     {
     *       expression: 'expression',
     *       locked_on_deployment: true,
     *       name: 'name',
     *       position: {},
     *       ref: 'ref',
     *       version: 0,
     *     },
     *   ],
     * });
     * ```
     */
    update(params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/environments`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List zone environments
     *
     * @example
     * ```ts
     * const environments = await client.zones.environments.list({
     *   zone_id: 'zone_id',
     * });
     * ```
     */
    list(params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/environments`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Delete zone environment
     *
     * @example
     * ```ts
     * const environment = await client.zones.environments.delete(
     *   'environment_id',
     *   { zone_id: 'zone_id' },
     * );
     * ```
     */
    delete(environmentId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/environments/${environmentId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Partially update zone environments
     *
     * @example
     * ```ts
     * const response = await client.zones.environments.edit({
     *   zone_id: 'zone_id',
     *   environments: [
     *     {
     *       expression: 'expression',
     *       locked_on_deployment: true,
     *       name: 'name',
     *       position: {},
     *       ref: 'ref',
     *       version: 0,
     *     },
     *   ],
     * });
     * ```
     */
    edit(params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/environments`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Roll back zone environment
     *
     * @example
     * ```ts
     * const response = await client.zones.environments.rollback(
     *   'environment_id',
     *   { zone_id: 'zone_id' },
     * );
     * ```
     */
    rollback(environmentId, params, options) {
        const { zone_id } = params;
        return this._client.post(`/zones/${zone_id}/environments/${environmentId}/rollback`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Environments = Environments;
