// cloudflare/resources/ddos-protection/advanced-tcp-protection/allowlist/allowlist.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as ItemsAPI from "./items.mjs";
import { Items, } from "./items.mjs";
import { V4PagePaginationArray } from "../../../../pagination.mjs";
export class Allowlist extends APIResource {
    constructor() {
        super(...arguments);
        this.items = new ItemsAPI.Items(this._client);
    }
    /**
     * Create an allowlist prefix for an account.
     *
     * @example
     * ```ts
     * const allowlist =
     *   await client.DDoSProtection.advancedTCPProtection.allowlist.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       comment: 'comment',
     *       enabled: true,
     *       prefix: 'prefix',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/allowlist`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all allowlist prefixes for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const allowlistListResponse of client.DDoSProtection.advancedTCPProtection.allowlist.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/allowlist`, AllowlistListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete all allowlist prefixes for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.DDoSProtection.advancedTCPProtection.allowlist.bulkDelete(
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkDelete(params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/magic/advanced_tcp_protection/configs/allowlist`, options);
    }
}
export class AllowlistListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Allowlist.AllowlistListResponsesV4PagePaginationArray = AllowlistListResponsesV4PagePaginationArray;
Allowlist.Items = Items;
