// cloudflare/resources/resource-tagging/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneTags = exports.Values = exports.ValueListResponsesCursorPaginationAfter = exports.ResourceTagging = exports.Keys = exports.KeyListResponsesCursorPaginationAfter = exports.AccountTags = void 0;
var account_tags_1 = require("./account-tags.js");
Object.defineProperty(exports, "AccountTags", { enumerable: true, get: function () { return account_tags_1.AccountTags; } });
var keys_1 = require("./keys.js");
Object.defineProperty(exports, "KeyListResponsesCursorPaginationAfter", { enumerable: true, get: function () { return keys_1.KeyListResponsesCursorPaginationAfter; } });
Object.defineProperty(exports, "Keys", { enumerable: true, get: function () { return keys_1.Keys; } });
var resource_tagging_1 = require("./resource-tagging.js");
Object.defineProperty(exports, "ResourceTagging", { enumerable: true, get: function () { return resource_tagging_1.ResourceTagging; } });
var values_1 = require("./values.js");
Object.defineProperty(exports, "ValueListResponsesCursorPaginationAfter", { enumerable: true, get: function () { return values_1.ValueListResponsesCursorPaginationAfter; } });
Object.defineProperty(exports, "Values", { enumerable: true, get: function () { return values_1.Values; } });
var zone_tags_1 = require("./zone-tags.js");
Object.defineProperty(exports, "ZoneTags", { enumerable: true, get: function () { return zone_tags_1.ZoneTags; } });
