// cloudflare/resources/resource-tagging/resource-tagging.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as AccountTagsAPI from "./account-tags.mjs";
import { AccountTags, } from "./account-tags.mjs";
import * as KeysAPI from "./keys.mjs";
import { KeyListResponsesCursorPaginationAfter, Keys } from "./keys.mjs";
import * as ValuesAPI from "./values.mjs";
import { ValueListResponsesCursorPaginationAfter, Values, } from "./values.mjs";
import * as ZoneTagsAPI from "./zone-tags.mjs";
import { ZoneTags, } from "./zone-tags.mjs";
import { CursorPaginationAfter } from "../../pagination.mjs";
export class ResourceTagging extends APIResource {
    constructor() {
        super(...arguments);
        this.accountTags = new AccountTagsAPI.AccountTags(this._client);
        this.zoneTags = new ZoneTagsAPI.ZoneTags(this._client);
        this.keys = new KeysAPI.Keys(this._client);
        this.values = new ValuesAPI.Values(this._client);
    }
    /**
     * Lists all tagged resources for an account.
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/tags/resources`, ResourceTaggingListResponsesCursorPaginationAfter, { query, ...options });
    }
}
export class ResourceTaggingListResponsesCursorPaginationAfter extends CursorPaginationAfter {
}
ResourceTagging.ResourceTaggingListResponsesCursorPaginationAfter =
    ResourceTaggingListResponsesCursorPaginationAfter;
ResourceTagging.AccountTags = AccountTags;
ResourceTagging.ZoneTags = ZoneTags;
ResourceTagging.Keys = Keys;
ResourceTagging.KeyListResponsesCursorPaginationAfter = KeyListResponsesCursorPaginationAfter;
ResourceTagging.Values = Values;
ResourceTagging.ValueListResponsesCursorPaginationAfter = ValueListResponsesCursorPaginationAfter;
