// cloudflare/resources/resource-tagging/resource-tagging.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceTaggingListResponsesCursorPaginationAfter = exports.ResourceTagging = void 0;
const resource_1 = require("../../resource.js");
const AccountTagsAPI = __importStar(require("./account-tags.js"));
const account_tags_1 = require("./account-tags.js");
const KeysAPI = __importStar(require("./keys.js"));
const keys_1 = require("./keys.js");
const ValuesAPI = __importStar(require("./values.js"));
const values_1 = require("./values.js");
const ZoneTagsAPI = __importStar(require("./zone-tags.js"));
const zone_tags_1 = require("./zone-tags.js");
const pagination_1 = require("../../pagination.js");
class ResourceTagging extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.accountTags = new AccountTagsAPI.AccountTags(this._client);
        this.zoneTags = new ZoneTagsAPI.ZoneTags(this._client);
        this.keys = new KeysAPI.Keys(this._client);
        this.values = new ValuesAPI.Values(this._client);
    }
    /**
     * Lists all tagged resources for an account.
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/tags/resources`, ResourceTaggingListResponsesCursorPaginationAfter, { query, ...options });
    }
}
exports.ResourceTagging = ResourceTagging;
class ResourceTaggingListResponsesCursorPaginationAfter extends pagination_1.CursorPaginationAfter {
}
exports.ResourceTaggingListResponsesCursorPaginationAfter = ResourceTaggingListResponsesCursorPaginationAfter;
ResourceTagging.ResourceTaggingListResponsesCursorPaginationAfter =
    ResourceTaggingListResponsesCursorPaginationAfter;
ResourceTagging.AccountTags = account_tags_1.AccountTags;
ResourceTagging.ZoneTags = zone_tags_1.ZoneTags;
ResourceTagging.Keys = keys_1.Keys;
ResourceTagging.KeyListResponsesCursorPaginationAfter = keys_1.KeyListResponsesCursorPaginationAfter;
ResourceTagging.Values = values_1.Values;
ResourceTagging.ValueListResponsesCursorPaginationAfter = values_1.ValueListResponsesCursorPaginationAfter;
