// cloudflare/resources/organizations/organizations.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import { isRequestOptions } from "../../core.mjs";
import * as OrganizationProfileAPI from "./organization-profile.mjs";
import { OrganizationProfileResource, } from "./organization-profile.mjs";
import * as LogsAPI from "./logs/logs.mjs";
import { Logs } from "./logs/logs.mjs";
import { SinglePage } from "../../pagination.mjs";
export class Organizations extends APIResource {
    constructor() {
        super(...arguments);
        this.organizationProfile = new OrganizationProfileAPI.OrganizationProfileResource(this._client);
        this.logs = new LogsAPI.Logs(this._client);
    }
    /**
     * Create a new organization for a user. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    create(body, options) {
        return this._client.post('/organizations', { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Modify organization. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    update(organizationId, body, options) {
        return this._client.put(`/organizations/${organizationId}`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    list(query = {}, options) {
        if (isRequestOptions(query)) {
            return this.list({}, query);
        }
        return this._client.getAPIList('/organizations', OrganizationsSinglePage, { query, ...options });
    }
    /**
     * Delete an organization. The organization MUST be empty before deleting. It must
     * not contain any sub-organizations, accounts, members or users. (Currently in
     * Closed Beta - see https://developers.cloudflare.com/fundamentals/organizations/)
     */
    delete(organizationId, options) {
        return this._client.delete(`/organizations/${organizationId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve the details of a certain organization. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    get(organizationId, options) {
        return this._client.get(`/organizations/${organizationId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class OrganizationsSinglePage extends SinglePage {
}
Organizations.OrganizationsSinglePage = OrganizationsSinglePage;
Organizations.OrganizationProfileResource = OrganizationProfileResource;
Organizations.Logs = Logs;
