// cloudflare/resources/organizations/organizations.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationsSinglePage = exports.Organizations = void 0;
const resource_1 = require("../../resource.js");
const core_1 = require("../../core.js");
const OrganizationProfileAPI = __importStar(require("./organization-profile.js"));
const organization_profile_1 = require("./organization-profile.js");
const LogsAPI = __importStar(require("./logs/logs.js"));
const logs_1 = require("./logs/logs.js");
const pagination_1 = require("../../pagination.js");
class Organizations extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.organizationProfile = new OrganizationProfileAPI.OrganizationProfileResource(this._client);
        this.logs = new LogsAPI.Logs(this._client);
    }
    /**
     * Create a new organization for a user. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    create(body, options) {
        return this._client.post('/organizations', { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Modify organization. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    update(organizationId, body, options) {
        return this._client.put(`/organizations/${organizationId}`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    list(query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.list({}, query);
        }
        return this._client.getAPIList('/organizations', OrganizationsSinglePage, { query, ...options });
    }
    /**
     * Delete an organization. The organization MUST be empty before deleting. It must
     * not contain any sub-organizations, accounts, members or users. (Currently in
     * Closed Beta - see https://developers.cloudflare.com/fundamentals/organizations/)
     */
    delete(organizationId, options) {
        return this._client.delete(`/organizations/${organizationId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve the details of a certain organization. (Currently in Closed Beta - see
     * https://developers.cloudflare.com/fundamentals/organizations/)
     */
    get(organizationId, options) {
        return this._client.get(`/organizations/${organizationId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Organizations = Organizations;
class OrganizationsSinglePage extends pagination_1.SinglePage {
}
exports.OrganizationsSinglePage = OrganizationsSinglePage;
Organizations.OrganizationsSinglePage = OrganizationsSinglePage;
Organizations.OrganizationProfileResource = organization_profile_1.OrganizationProfileResource;
Organizations.Logs = logs_1.Logs;
