// cloudflare/resources/browser-rendering/devtools/browser/browser.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Browser = void 0;
const resource_1 = require("../../../../resource.js");
const PageAPI = __importStar(require("./page.js"));
const page_1 = require("./page.js");
const TargetsAPI = __importStar(require("./targets.js"));
const targets_1 = require("./targets.js");
class Browser extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.page = new PageAPI.Page(this._client);
        this.targets = new TargetsAPI.Targets(this._client);
    }
    /**
     * Get a browser session ID.
     *
     * @example
     * ```ts
     * const browser =
     *   await client.browserRendering.devtools.browser.create({
     *     account_id: 'account_id',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, keep_alive, lab, recording, targets } = params;
        return this._client.post(`/accounts/${account_id}/browser-rendering/devtools/browser`, {
            query: { keep_alive, lab, recording, targets },
            ...options,
        });
    }
    /**
     * Closes an existing browser session.
     *
     * @example
     * ```ts
     * const browser =
     *   await client.browserRendering.devtools.browser.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(sessionId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}`, options);
    }
    /**
     * Establishes a WebSocket connection to an existing browser session.
     *
     * @example
     * ```ts
     * await client.browserRendering.devtools.browser.connect(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { account_id: 'account_id' },
     * );
     * ```
     */
    connect(sessionId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}`, {
            query,
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Acquires and establishes a WebSocket connection to a browser session.
     *
     * @example
     * ```ts
     * await client.browserRendering.devtools.browser.launch({
     *   account_id: 'account_id',
     * });
     * ```
     */
    launch(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser`, {
            query,
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Returns the complete Chrome DevTools Protocol schema including all domains,
     * commands, events, and types. This schema describes the entire CDP API surface.
     *
     * @example
     * ```ts
     * const response =
     *   await client.browserRendering.devtools.browser.protocol(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    protocol(sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}/json/protocol`, options);
    }
    /**
     * Get browser version metadata.
     *
     * @example
     * ```ts
     * const response =
     *   await client.browserRendering.devtools.browser.version(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    version(sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}/json/version`, options);
    }
}
exports.Browser = Browser;
Browser.Page = page_1.Page;
Browser.Targets = targets_1.Targets;
