// cloudflare/resources/browser-rendering/devtools/browser/browser.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../resource.mjs";
import * as PageAPI from "./page.mjs";
import { Page } from "./page.mjs";
import * as TargetsAPI from "./targets.mjs";
import { Targets, } from "./targets.mjs";
export class Browser extends APIResource {
    constructor() {
        super(...arguments);
        this.page = new PageAPI.Page(this._client);
        this.targets = new TargetsAPI.Targets(this._client);
    }
    /**
     * Get a browser session ID.
     *
     * @example
     * ```ts
     * const browser =
     *   await client.browserRendering.devtools.browser.create({
     *     account_id: 'account_id',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, keep_alive, lab, recording, targets } = params;
        return this._client.post(`/accounts/${account_id}/browser-rendering/devtools/browser`, {
            query: { keep_alive, lab, recording, targets },
            ...options,
        });
    }
    /**
     * Closes an existing browser session.
     *
     * @example
     * ```ts
     * const browser =
     *   await client.browserRendering.devtools.browser.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(sessionId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}`, options);
    }
    /**
     * Establishes a WebSocket connection to an existing browser session.
     *
     * @example
     * ```ts
     * await client.browserRendering.devtools.browser.connect(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { account_id: 'account_id' },
     * );
     * ```
     */
    connect(sessionId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}`, {
            query,
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Acquires and establishes a WebSocket connection to a browser session.
     *
     * @example
     * ```ts
     * await client.browserRendering.devtools.browser.launch({
     *   account_id: 'account_id',
     * });
     * ```
     */
    launch(params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser`, {
            query,
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Returns the complete Chrome DevTools Protocol schema including all domains,
     * commands, events, and types. This schema describes the entire CDP API surface.
     *
     * @example
     * ```ts
     * const response =
     *   await client.browserRendering.devtools.browser.protocol(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    protocol(sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}/json/protocol`, options);
    }
    /**
     * Get browser version metadata.
     *
     * @example
     * ```ts
     * const response =
     *   await client.browserRendering.devtools.browser.version(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    version(sessionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/browser-rendering/devtools/browser/${sessionId}/json/version`, options);
    }
}
Browser.Page = Page;
Browser.Targets = Targets;
