// cloudflare/resources/content-scanning/content-scanning.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as PayloadsAPI from "./payloads.mjs";
import { PayloadCreateResponsesSinglePage, PayloadDeleteResponsesSinglePage, PayloadListResponsesSinglePage, Payloads, } from "./payloads.mjs";
import * as SettingsAPI from "./settings.mjs";
import { Settings } from "./settings.mjs";
export class ContentScanning extends APIResource {
    constructor() {
        super(...arguments);
        this.payloads = new PayloadsAPI.Payloads(this._client);
        this.settings = new SettingsAPI.Settings(this._client);
    }
    /**
     * Update the Content Scanning status.
     *
     * @example
     * ```ts
     * const contentScanning = await client.contentScanning.create(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     value: 'enabled',
     *   },
     * );
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/content-upload-scan/settings`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update the Content Scanning status.
     *
     * @example
     * ```ts
     * const contentScanning = await client.contentScanning.update(
     *   {
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     value: 'enabled',
     *   },
     * );
     * ```
     */
    update(params, options) {
        const { zone_id, ...body } = params;
        return this._client.put(`/zones/${zone_id}/content-upload-scan/settings`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Disable Content Scanning.
     *
     * @example
     * ```ts
     * const response = await client.contentScanning.disable({
     *   zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     * });
     * ```
     */
    disable(params, options) {
        const { zone_id } = params;
        return this._client.post(`/zones/${zone_id}/content-upload-scan/disable`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Enable Content Scanning.
     *
     * @example
     * ```ts
     * const response = await client.contentScanning.enable({
     *   zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     * });
     * ```
     */
    enable(params, options) {
        const { zone_id } = params;
        return this._client.post(`/zones/${zone_id}/content-upload-scan/enable`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieve the current status of Content Scanning.
     *
     * @example
     * ```ts
     * const contentScanning = await client.contentScanning.get({
     *   zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     * });
     * ```
     */
    get(params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/content-upload-scan/settings`, options)._thenUnwrap((obj) => obj.result);
    }
}
ContentScanning.Payloads = Payloads;
ContentScanning.PayloadCreateResponsesSinglePage = PayloadCreateResponsesSinglePage;
ContentScanning.PayloadListResponsesSinglePage = PayloadListResponsesSinglePage;
ContentScanning.PayloadDeleteResponsesSinglePage = PayloadDeleteResponsesSinglePage;
ContentScanning.Settings = Settings;
