// cloudflare/resources/ai-gateway/dynamic-routing.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
export class DynamicRouting extends APIResource {
    /**
     * Create a new AI Gateway Dynamic Route.
     *
     * @example
     * ```ts
     * const dynamicRouting =
     *   await client.aiGateway.dynamicRouting.create('54442216', {
     *     account_id: '0d37909e38d3e99c29fa2cd343ac421a',
     *     elements: [
     *       {
     *         id: 'id',
     *         outputs: { next: { elementId: 'elementId' } },
     *         type: 'start',
     *       },
     *     ],
     *     name: 'name',
     *   });
     * ```
     */
    create(gatewayId, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update an AI Gateway Dynamic Route.
     *
     * @example
     * ```ts
     * const dynamicRouting =
     *   await client.aiGateway.dynamicRouting.update(
     *     '54442216',
     *     '54442216',
     *     {
     *       account_id: '0d37909e38d3e99c29fa2cd343ac421a',
     *       name: 'Route Name',
     *     },
     *   );
     * ```
     */
    update(gatewayId, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}`, {
            body,
            ...options,
        });
    }
    /**
     * List all AI Gateway Dynamic Routes.
     *
     * @example
     * ```ts
     * const dynamicRoutings =
     *   await client.aiGateway.dynamicRouting.list('54442216', {
     *     account_id: '0d37909e38d3e99c29fa2cd343ac421a',
     *   });
     * ```
     */
    list(gatewayId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes`, {
            query,
            ...options,
        });
    }
    /**
     * Delete an AI Gateway Dynamic Route.
     *
     * @example
     * ```ts
     * const dynamicRouting =
     *   await client.aiGateway.dynamicRouting.delete(
     *     '54442216',
     *     '54442216',
     *     { account_id: '0d37909e38d3e99c29fa2cd343ac421a' },
     *   );
     * ```
     */
    delete(gatewayId, id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create a new AI Gateway Dynamic Route Deployment.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.dynamicRouting.createDeployment(
     *     '54442216',
     *     '54442216',
     *     {
     *       account_id: '0d37909e38d3e99c29fa2cd343ac421a',
     *       version_id: '54442216',
     *     },
     *   );
     * ```
     */
    createDeployment(gatewayId, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}/deployments`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create a new AI Gateway Dynamic Route Version.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.dynamicRouting.createVersion(
     *     '54442216',
     *     '54442216',
     *     {
     *       account_id: '0d37909e38d3e99c29fa2cd343ac421a',
     *       elements: [
     *         {
     *           id: 'id',
     *           outputs: { next: { elementId: 'elementId' } },
     *           type: 'start',
     *         },
     *       ],
     *     },
     *   );
     * ```
     */
    createVersion(gatewayId, id, params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}/versions`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get an AI Gateway Dynamic Route.
     *
     * @example
     * ```ts
     * const dynamicRouting =
     *   await client.aiGateway.dynamicRouting.get(
     *     '54442216',
     *     '54442216',
     *     { account_id: '0d37909e38d3e99c29fa2cd343ac421a' },
     *   );
     * ```
     */
    get(gatewayId, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get an AI Gateway Dynamic Route Version.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.dynamicRouting.getVersion(
     *     '54442216',
     *     '54442216',
     *     '54442216',
     *     { account_id: '0d37909e38d3e99c29fa2cd343ac421a' },
     *   );
     * ```
     */
    getVersion(gatewayId, id, versionId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}/versions/${versionId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all AI Gateway Dynamic Route Deployments.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.dynamicRouting.listDeployments(
     *     '54442216',
     *     '54442216',
     *     { account_id: '0d37909e38d3e99c29fa2cd343ac421a' },
     *   );
     * ```
     */
    listDeployments(gatewayId, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}/deployments`, options);
    }
    /**
     * List all AI Gateway Dynamic Route Versions.
     *
     * @example
     * ```ts
     * const response =
     *   await client.aiGateway.dynamicRouting.listVersions(
     *     '54442216',
     *     '54442216',
     *     { account_id: '0d37909e38d3e99c29fa2cd343ac421a' },
     *   );
     * ```
     */
    listVersions(gatewayId, id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/ai-gateway/gateways/${gatewayId}/routes/${id}/versions`, options);
    }
}
