// cloudflare/resources/zero-trust/access/ai-controls/mcp/servers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../../resource.mjs";
import { V4PagePaginationArray } from "../../../../../pagination.mjs";
export class Servers extends APIResource {
    /**
     * Creates a new MCP portal for managing AI tool access through Cloudflare Access.
     *
     * @example
     * ```ts
     * const server =
     *   await client.zeroTrust.access.aiControls.mcp.servers.create(
     *     {
     *       account_id: 'a86a8f5c339544d7bdc89926de14fb8c',
     *       id: 'my-mcp-server',
     *       auth_type: 'unauthenticated',
     *       hostname: 'https://example.com/mcp',
     *       name: 'My MCP Server',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/access/ai-controls/mcp/servers`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates an MCP portal configuration.
     *
     * @example
     * ```ts
     * const server =
     *   await client.zeroTrust.access.aiControls.mcp.servers.update(
     *     'my-mcp-server',
     *     { account_id: 'a86a8f5c339544d7bdc89926de14fb8c' },
     *   );
     * ```
     */
    update(id, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/access/ai-controls/mcp/servers/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all MCP portals configured for the account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const serverListResponse of client.zeroTrust.access.aiControls.mcp.servers.list(
     *   { account_id: 'a86a8f5c339544d7bdc89926de14fb8c' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/access/ai-controls/mcp/servers`, ServerListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes an MCP portal from the account.
     *
     * @example
     * ```ts
     * const server =
     *   await client.zeroTrust.access.aiControls.mcp.servers.delete(
     *     'my-mcp-server',
     *     { account_id: 'a86a8f5c339544d7bdc89926de14fb8c' },
     *   );
     * ```
     */
    delete(id, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/access/ai-controls/mcp/servers/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves gateway configuration for MCP portals.
     *
     * @example
     * ```ts
     * const response =
     *   await client.zeroTrust.access.aiControls.mcp.servers.read(
     *     'my-mcp-server',
     *     { account_id: 'a86a8f5c339544d7bdc89926de14fb8c' },
     *   );
     * ```
     */
    read(id, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/access/ai-controls/mcp/servers/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Syncs an MCP server's tool catalog with the portal.
     *
     * @example
     * ```ts
     * const response =
     *   await client.zeroTrust.access.aiControls.mcp.servers.sync(
     *     'my-mcp-portal',
     *     { account_id: 'a86a8f5c339544d7bdc89926de14fb8c' },
     *   );
     * ```
     */
    sync(id, params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/access/ai-controls/mcp/servers/${id}/sync`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class ServerListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Servers.ServerListResponsesV4PagePaginationArray = ServerListResponsesV4PagePaginationArray;
