// cloudflare/resources/token-validation/configuration/configuration.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as CredentialsAPI from "./credentials.mjs";
import { Credentials as CredentialsAPICredentials, } from "./credentials.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class Configuration extends APIResource {
    constructor() {
        super(...arguments);
        this.credentials = new CredentialsAPI.Credentials(this._client);
    }
    /**
     * Create a new Token Validation configuration
     *
     * @example
     * ```ts
     * const tokenConfig =
     *   await client.tokenValidation.configuration.create({
     *     zone_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     credentials: {
     *       keys: [
     *         {
     *           alg: 'ES256',
     *           crv: 'P-256',
     *           kid: '38013f13-c266-4eec-a72a-92ec92779f21',
     *           kty: 'EC',
     *           x: 'KN53JRwN3wCjm2o39bvZUX2VdrsHzS8pxOAGjm8m7EQ',
     *           y: 'lnkkzIxaveggz-HFhcMWW15nxvOj0Z_uQsXbpK0GFcY',
     *         },
     *       ],
     *     },
     *     description:
     *       'Long description for Token Validation Configuration',
     *     title: 'Example Token Validation Configuration',
     *     token_sources: [
     *       'http.request.headers["x-auth"][0]',
     *       'http.request.cookies["Authorization"][0]',
     *     ],
     *     token_type: 'JWT',
     *   });
     * ```
     */
    create(params, options) {
        const { zone_id, ...body } = params;
        return this._client.post(`/zones/${zone_id}/token_validation/config`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all token validation configurations for this zone
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const tokenConfig of client.tokenValidation.configuration.list(
     *   { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { zone_id, ...query } = params;
        return this._client.getAPIList(`/zones/${zone_id}/token_validation/config`, TokenConfigsV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete Token Configuration
     *
     * @example
     * ```ts
     * const configuration =
     *   await client.tokenValidation.configuration.delete(
     *     '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(configId, params, options) {
        const { zone_id } = params;
        return this._client.delete(`/zones/${zone_id}/token_validation/config/${configId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Edit fields of an existing Token Configuration
     *
     * @example
     * ```ts
     * const response =
     *   await client.tokenValidation.configuration.edit(
     *     '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    edit(configId, params, options) {
        const { zone_id, ...body } = params;
        return this._client.patch(`/zones/${zone_id}/token_validation/config/${configId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a single Token Configuration
     *
     * @example
     * ```ts
     * const tokenConfig =
     *   await client.tokenValidation.configuration.get(
     *     '4a7ee8d3-dd63-4ceb-9d5f-c27831854ce7',
     *     { zone_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(configId, params, options) {
        const { zone_id } = params;
        return this._client.get(`/zones/${zone_id}/token_validation/config/${configId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class TokenConfigsV4PagePaginationArray extends V4PagePaginationArray {
}
Configuration.TokenConfigsV4PagePaginationArray = TokenConfigsV4PagePaginationArray;
Configuration.Credentials = CredentialsAPICredentials;
