// cloudflare/resources/email-security/settings/block-senders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class BlockSenders extends APIResource {
    /**
     * Creates a new blocked sender pattern. Emails matching this pattern will be
     * blocked from delivery. Patterns can be email addresses, domains, or IP
     * addresses, and support regular expressions.
     *
     * @example
     * ```ts
     * const blockSender =
     *   await client.emailSecurity.settings.blockSenders.create({
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     is_regex: false,
     *     pattern: 'test@example.com',
     *     pattern_type: 'EMAIL',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/email-security/settings/block_senders`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns a paginated list of blocked email sender patterns. These patterns
     * prevent emails from matching senders from being delivered. Supports filtering by
     * pattern type and searching across patterns.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const blockSenderListResponse of client.emailSecurity.settings.blockSenders.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/email-security/settings/block_senders`, BlockSenderListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Removes a blocked sender pattern. After deletion, emails from this sender will
     * no longer be automatically blocked based on this rule.
     *
     * @example
     * ```ts
     * const blockSender =
     *   await client.emailSecurity.settings.blockSenders.delete(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(patternId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/email-security/settings/block_senders/${patternId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates an existing blocked sender pattern. Only provided fields will be
     * modified. The pattern will continue blocking emails until deleted.
     *
     * @example
     * ```ts
     * const response =
     *   await client.emailSecurity.settings.blockSenders.edit(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    edit(patternId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/email-security/settings/block_senders/${patternId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves details for a specific blocked sender pattern including its pattern
     * type, value, and metadata.
     *
     * @example
     * ```ts
     * const blockSender =
     *   await client.emailSecurity.settings.blockSenders.get(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(patternId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/email-security/settings/block_senders/${patternId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class BlockSenderListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
BlockSenders.BlockSenderListResponsesV4PagePaginationArray = BlockSenderListResponsesV4PagePaginationArray;
