// cloudflare/resources/email-security/settings/trusted-domains.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustedDomainListResponsesV4PagePaginationArray = exports.TrustedDomains = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class TrustedDomains extends resource_1.APIResource {
    /**
     * Creates a new trusted domain pattern. Use for partner domains or approved
     * senders that should bypass recent domain registration and similarity checks.
     * Configure whether it prevents recent domain or spoof dispositions.
     *
     * @example
     * ```ts
     * const trustedDomain =
     *   await client.emailSecurity.settings.trustedDomains.create(
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       is_recent: true,
     *       is_regex: false,
     *       is_similarity: false,
     *       pattern: 'example.com',
     *     },
     *   );
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/email-security/settings/trusted_domains`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns a paginated list of trusted domain patterns. Trusted domains prevent
     * false positives for recently registered domains and lookalike domain detections.
     * Patterns can use regular expressions for flexible matching.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const trustedDomainListResponse of client.emailSecurity.settings.trustedDomains.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/email-security/settings/trusted_domains`, TrustedDomainListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Removes a trusted domain pattern. After deletion, emails from this domain will
     * be subject to normal recent domain and similarity checks.
     *
     * @example
     * ```ts
     * const trustedDomain =
     *   await client.emailSecurity.settings.trustedDomains.delete(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(trustedDomainId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/email-security/settings/trusted_domains/${trustedDomainId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates an existing trusted domain pattern. Only provided fields will be
     * modified. Changes take effect for new emails matching the pattern.
     *
     * @example
     * ```ts
     * const response =
     *   await client.emailSecurity.settings.trustedDomains.edit(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    edit(trustedDomainId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/email-security/settings/trusted_domains/${trustedDomainId}`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves details for a specific trusted domain pattern including its pattern
     * value, whether it uses regex matching, and which detection types it affects.
     *
     * @example
     * ```ts
     * const trustedDomain =
     *   await client.emailSecurity.settings.trustedDomains.get(
     *     'f174e90a-fafe-4643-bbbc-4a0ed4fc8415',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(trustedDomainId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/email-security/settings/trusted_domains/${trustedDomainId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.TrustedDomains = TrustedDomains;
class TrustedDomainListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.TrustedDomainListResponsesV4PagePaginationArray = TrustedDomainListResponsesV4PagePaginationArray;
TrustedDomains.TrustedDomainListResponsesV4PagePaginationArray =
    TrustedDomainListResponsesV4PagePaginationArray;
