// cloudflare/resources/connectivity/directory/services.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { V4PagePaginationArray } from "../../../pagination.mjs";
export class Services extends APIResource {
    /**
     * Create Workers VPC connectivity service
     *
     * @example
     * ```ts
     * const service =
     *   await client.connectivity.directory.services.create({
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     host: {
     *       hostname: 'api.example.com',
     *       resolver_network: {
     *         tunnel_id: '0191dce4-9ab4-7fce-b660-8e5dec5172da',
     *       },
     *     },
     *     name: 'web-server',
     *     type: 'http',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/connectivity/directory/services`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update Workers VPC connectivity service
     *
     * @example
     * ```ts
     * const service =
     *   await client.connectivity.directory.services.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       account_id: 'account_id',
     *       host: {
     *         ipv4: '10.0.0.1',
     *         network: {
     *           tunnel_id: '0191dce4-9ab4-7fce-b660-8e5dec5172da',
     *         },
     *       },
     *       name: 'postgres-db',
     *       type: 'tcp',
     *     },
     *   );
     * ```
     */
    update(serviceId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/connectivity/directory/services/${serviceId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List Workers VPC connectivity services
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const serviceListResponse of client.connectivity.directory.services.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/connectivity/directory/services`, ServiceListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete Workers VPC connectivity service
     *
     * @example
     * ```ts
     * await client.connectivity.directory.services.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { account_id: 'account_id' },
     * );
     * ```
     */
    delete(serviceId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/connectivity/directory/services/${serviceId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Get Workers VPC connectivity service
     *
     * @example
     * ```ts
     * const service =
     *   await client.connectivity.directory.services.get(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    get(serviceId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/connectivity/directory/services/${serviceId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class ServiceListResponsesV4PagePaginationArray extends V4PagePaginationArray {
}
Services.ServiceListResponsesV4PagePaginationArray = ServiceListResponsesV4PagePaginationArray;
