// cloudflare/resources/radar/netflows/netflows.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { isRequestOptions } from "../../../core.mjs";
import * as TopAPI from "./top.mjs";
import { Top } from "./top.mjs";
export class NetFlows extends APIResource {
    constructor() {
        super(...arguments);
        this.top = new TopAPI.Top(this._client);
    }
    summary(query = {}, options) {
        if (isRequestOptions(query)) {
            return this.summary({}, query);
        }
        return this._client.get('/radar/netflows/summary', { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    summaryV2(dimension, query = {}, options) {
        if (isRequestOptions(query)) {
            return this.summaryV2(dimension, {}, query);
        }
        return this._client.get(`/radar/netflows/summary/${dimension}`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    timeseries(query = {}, options) {
        if (isRequestOptions(query)) {
            return this.timeseries({}, query);
        }
        return this._client.get('/radar/netflows/timeseries', { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    timeseriesGroups(dimension, query = {}, options) {
        if (isRequestOptions(query)) {
            return this.timeseriesGroups(dimension, {}, query);
        }
        return this._client.get(`/radar/netflows/timeseries_groups/${dimension}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
NetFlows.Top = Top;
